---
title: "Generate the supplementary table to explain the disconcordance reason"
output: html_notebook
---

This is the script to generate the supplementary table of all mutations (SNVs) in GDC and MC3 public calls and assign reasons to individual mutations why they did not appear in the public call set of the other group.

In the supplementary table, we only include the following columns:

- chromosome
- start_position
- end_position
- cancer_type
- tumor_sample_barcode
- reference_allele
- tumor_seq_allele_1
- tumor_seq_allele_2
- hgvsc
- hgvsp
- variant_classification
- variant_type
- hugo_symbol
- transcript_id
- concordance_status
- reason_not_included_in_public_mc3
- reason_not_included_in_public_gdc

Readers should be able to pin point the same mutation call in the original files released by both groups using the provided columns in the table.

Load necessary packages:
```{r}
library(tidyverse)
library(DBI)
library(RSQLite)
```

Connect to the database
```{r}
conn = dbConnect(RSQLite::SQLite(), dbname='../processed_data/all_variants.sqlite')
dbExecute(conn, 'PRAGMA cache_size=-4000000')
dbExecute(conn, 'PRAGMA temp_store=MEMORY')
dbExecute(conn, 'PRAGMA query_only=1')
dbListTables(conn)
```

Load all public SNV calls.
```{r}
all_public_snv_tbl <- dbReadTable(conn, 'full_overlap') %>%
    as.tibble() %>%
    filter(mc3_variant_type == 'SNP' | gdc_variant_type == 'SNP')
```


## Concordant calls (shared by GDC and MC3)
These SNV calls were shared by GDC and MC3, so we just add it directly to the supplementary table. We prefer the information provided by GDC.
```{r}
supp_tbl <- all_public_snv_tbl %>%
    filter(shared_by_gdc_mc3 == 1) %>%
    select(
        chromosome, start_position, end_position, tumor_sample_barcode, cancer_type,
        reference_allele, tumor_seq_allele1=gdc_tumor_seq_allele1, tumor_seq_allele2,
        hgvsc=gdc_hgvsc, hgvsp=gdc_hgvsp, 
        variant_classification=gdc_variant_classification, variant_type=gdc_variant_type,
        hugo_symbol=gdc_hugo_symbol, transcript_id=gdc_transcript_id
    ) %>%
    # Add additional columns
    mutate(
        concordance_status = 'shared',
        reason_not_included_in_public_gdc = NA_character_,
        reason_not_included_in_public_mc3 = NA_character_
    )

supp_tbl %>% head()
```


## Unrecoverable unique calls
These calls are the mutation calls only reported by the callers of one group but not in the other group.
```{r}
gdc_unrecoverable_unique_tbl <- read_tsv(
    '../processed_data/gdc_not_recoverable_unique_variants.tsv.gz',
    col_types = cols(
        .default = col_character(),
        start_position = col_integer(), end_position = col_integer(),
        mc3_t_depth = col_integer(), mc3_t_ref_count = col_integer(), mc3_t_alt_count = col_integer(),
        mc3_n_depth = col_integer(), mc3_n_ref_count = col_integer(), mc3_n_alt_count = col_integer()
    ),
    progress = FALSE
) %>% filter(gdc_variant_type == 'SNP')

mc3_unrecoverable_unique_tbl <- read_tsv(
    '../processed_data/mc3_not_recoverable_unique_variants.tsv.gz',
    col_types = cols(
        .default = col_character(),
        start_position = col_integer(), end_position = col_integer(),
        mc3_t_depth = col_integer(), mc3_t_ref_count = col_integer(), mc3_t_alt_count = col_integer(),
        mc3_n_depth = col_integer(), mc3_n_ref_count = col_integer(), mc3_n_alt_count = col_integer()
    ),
    progress = FALSE
) %>% filter(mc3_variant_type == 'SNP')
```

Unrecoverable GDC-unique calls are only called in gdc. They could be not called at all in MC3, or they were of bad quality (pre-filtering) in MC3.
```{r}
current_mut_tbl <- gdc_unrecoverable_unique_tbl %>%
     select(
        chromosome, start_position, end_position, tumor_sample_barcode, cancer_type,
        reference_allele, tumor_seq_allele1=gdc_tumor_seq_allele1, tumor_seq_allele2,
        hgvsc=gdc_hgvsc, hgvsp=gdc_hgvsp, 
        variant_classification=gdc_variant_classification, variant_type=gdc_variant_type,
        hugo_symbol=gdc_hugo_symbol, transcript_id=gdc_transcript_id
    ) %>%
    # Add additional columns
    mutate(
        concordance_status = 'unrecoverable GDC-unique',
        reason_not_included_in_public_gdc = NA_character_,
        reason_not_included_in_public_mc3 = 'not called or bad quality in MC3'
    )

current_mut_tbl %>% head()
```

Add them into the supplementary table.
```{r}
supp_tbl <- bind_rows(supp_tbl, current_mut_tbl)
```


Unrecoverable MC3-unique calls are only called in MC3. 
```{r}
current_mut_tbl <- mc3_unrecoverable_unique_tbl %>%
     select(
        chromosome, start_position, end_position, tumor_sample_barcode, cancer_type,
        reference_allele, tumor_seq_allele1=mc3_tumor_seq_allele1, tumor_seq_allele2,
        hgvsc=mc3_hgvsc, hgvsp=mc3_hgvsp, 
        variant_classification=mc3_variant_classification, variant_type=mc3_variant_type,
        hugo_symbol=mc3_hugo_symbol, transcript_id=mc3_transcript_id
    ) %>%
    # Add additional columns
    mutate(
        concordance_status = 'unrecoverable MC3-unique',
        reason_not_included_in_public_gdc = 'not called in GDC',
        reason_not_included_in_public_mc3 = NA_character_
    )

current_mut_tbl %>% head()
```

Add them into the supplementary table.
```{r}
supp_tbl <- bind_rows(supp_tbl, current_mut_tbl)
```



## Recoverable unique calls 
These calls only appear in the public call of one group, but they could be found in the pre-filter calls (protected calls) of the other group. This implies they were excluded during the filtering. Here we assign the reason why each call was filtered.
Read recoverable unique calls.

```{r}
gdc_unique_recoverable_tbl <- read_tsv(
    '../processed_data/gdc_recoverable_unique_variants.tsv.gz',
    col_types = cols(
        .default = col_character(),
        start_position = col_integer(), end_position = col_integer(),
        mc3_t_depth = col_integer(), mc3_t_ref_count = col_integer(), mc3_t_alt_count = col_integer(),
        mc3_n_depth = col_integer(), mc3_n_ref_count = col_integer(), mc3_n_alt_count = col_integer(),
        overlap_rowid = col_integer(), gdc_rowid = col_integer(), mc3_protected_rowid = col_integer()
    ),
    progress = FALSE
)
gdc_filters_tbl <- read_tsv(
    '../processed_data/gdc_recoverable_unique_variants.filter_cols.tsv.gz',
    progress = FALSE
) 
mc3_unique_recoverable_tbl <- read_tsv(
    '../processed_data/mc3_recoverable_unique_variants.tsv.gz',
        col_types = cols(
        .default = col_character(),
        start_position = col_integer(), end_position = col_integer(),
        mc3_t_depth = col_integer(), mc3_t_ref_count = col_integer(), mc3_t_alt_count = col_integer(),
        mc3_n_depth = col_integer(), mc3_n_ref_count = col_integer(), mc3_n_alt_count = col_integer(),
        overlap_rowid = col_integer(), mc3_rowid = col_integer(), gdc_protected_rowid = col_integer()
    ),
    progress = FALSE
)
mc3_filters_tbl <- read_tsv(
    '../processed_data/mc3_recoverable_unique_variants.filter_cols.tsv.gz',
    progress = FALSE
)

# Merge with its filter columns
gdc_tbl <- bind_cols(gdc_unique_recoverable_tbl, gdc_filters_tbl) %>%
    rename(gdc_filter_raw = gdc_filter, gdc_gdc_filter_raw = gdc_gdc_filter,
           gdc_filter = gdc_filter_unique, gdc_gdc_filter = gdc_gdc_filter_unique) %>%
    filter(gdc_variant_type == 'SNP')
mc3_tbl <- bind_cols(mc3_unique_recoverable_tbl, mc3_filters_tbl) %>%
    rename(gdc_filter_raw = gdc_filter, gdc_gdc_filter_raw = gdc_gdc_filter,
       gdc_filter = gdc_filter_unique, gdc_gdc_filter = gdc_gdc_filter_unique) %>%
    filter(mc3_variant_type == 'SNP')
```

Since one mutation call may appear multiple times in the protected calls of two groups if different callers reported different allele frequence (which would be two calls), we de-duplicate the records back.
```{r}
gdc_tbl <- gdc_tbl %>% distinct(gdc_rowid, .keep_all = TRUE)
mc3_tbl <- mc3_tbl %>% distinct(mc3_rowid, .keep_all = TRUE)
```

Due to some missing columns in `gdc_tbl` and `mc3_tbl`, in this section, we use the `gdc_rowid` or `mc3_rowid` of each mutation call to retrieve their full records from the `all_public_snv_tbl` table.

```{r}
retreive_full_record_via_gdc <- function(mut_tbl) {
    mut_tbl %>%
        select(gdc_rowid) %>%
        left_join(all_public_snv_tbl, by = 'gdc_rowid') %>%
        select(
            gdc_rowid,
            chromosome, start_position, end_position, tumor_sample_barcode, cancer_type,
            reference_allele, tumor_seq_allele1=gdc_tumor_seq_allele1, tumor_seq_allele2,
            hgvsc=gdc_hgvsc, hgvsp=gdc_hgvsp, 
            variant_classification=gdc_variant_classification, variant_type=gdc_variant_type,
            hugo_symbol=gdc_hugo_symbol, transcript_id=gdc_transcript_id
        )
}
```



## Recoverable GDC-unique calls

### One-caller constraint
The major reason a mutation call was filtered in MC3 was the one-caller constraint.
```{r}
current_mut_tbl <- gdc_tbl %>% 
    filter(mc3_ncallers == 1) %>% 
    # Select only the rowid as the information will be retrieved from the public SNP table
    retreive_full_record_via_gdc() %>%
    # Add additional columns
    mutate(
        concordance_status = 'recoverable GDC-unique',
        reason_not_included_in_public_gdc = NA_character_,
        reason_not_included_in_public_mc3 = 'called by only one caller'
    )

current_mut_tbl %>% head()
```

Double check the record is the same
```{r}
gdc_tbl %>% 
    filter(mc3_ncallers == 1) %>%
    select(
        chromosome, start_position, end_position, tumor_sample_barcode, cancer_type
    ) %>%
    head()
```

Add them into the supplementary table. And exclude the explained calls.
```{r}
supp_tbl <- bind_rows(supp_tbl, current_mut_tbl %>% select(-gdc_rowid))
gdc_tbl <- gdc_tbl %>% filter(!gdc_rowid %in% current_mut_tbl$gdc_rowid)
```

### NonExonic filter definition
The second reason is the different definition of Non-Exonic filter in GDC and MC3. As long as there is one variant effect that is exonic, GDC will keep the call. On the other hand, MC3 will drop the call if the main variant effect is non-exonic.
```{r}
current_mut_tbl <- gdc_tbl %>% 
    # The main effect in MC3 is non-exonic
    filter(mc3__NonExonic == 1) %>%
    # The main effect in GDC is also non-exonic, but one of the possible effect is exonic
    filter(
        gdc_variant_classification %in% c('Intron', 'RNA', "5'Flank", "3'Flank", "IGR", 'Splice_Region', 'Splice_Site'),
        str_detect(gdc_all_effects, 'exon_variant') 
        | str_detect(gdc_all_effects, 'missense_variant') 
        | str_detect(gdc_all_effects, 'prime_UTR_variant')
        | str_detect(gdc_all_effects, 'synonymous_variant')
        | str_detect(gdc_all_effects, 'frameshift_variant')
    ) %>%
    # Select only the rowid as the information will be retrieved from the public SNP table
    retreive_full_record_via_gdc() %>%
    # Add additional columns
    mutate(
        concordance_status = 'recoverable GDC-unique',
        reason_not_included_in_public_gdc = NA_character_,
        reason_not_included_in_public_mc3 = 'filtered by NonExonic (different definition)'
    )

nrow(current_mut_tbl)
```

Add them into the supplementary table. And exclude the explained calls.
```{r}
supp_tbl <- bind_rows(supp_tbl, current_mut_tbl %>% select(-gdc_rowid))
gdc_tbl <- gdc_tbl %>% filter(!gdc_rowid %in% current_mut_tbl$gdc_rowid)
```

### Use of validation sequencing
GDC will promote the call if it can be detected in the validation sequencing
```{r}
current_mut_tbl <- gdc_tbl %>% 
    # The main effect in MC3 is non-exonic
    filter(mc3__NonExonic == 1) %>%
    # The main effect in GDC is non-exonic, and all effects are non-exonic
    filter(
        gdc_variant_classification %in% c('Intron', 'RNA', "5'Flank", "3'Flank", "IGR", 'Splice_Region', 'Splice_Site'),
        !(str_detect(gdc_all_effects, 'exon_variant') 
        | str_detect(gdc_all_effects, 'missense_variant') 
        | str_detect(gdc_all_effects, 'prime_UTR_variant')
        | str_detect(gdc_all_effects, 'synonymous_variant')
        | str_detect(gdc_all_effects, 'frameshift_variant'))
    ) %>%
    filter(str_detect(gdc_validation_status, 'Valid'))%>%
    retreive_full_record_via_gdc() %>%
    mutate(
        concordance_status = 'recoverable GDC-unique',
        reason_not_included_in_public_gdc = NA_character_,
        reason_not_included_in_public_mc3 = 'filtered by NonExonic (detected in validation sequencing)'
    )

nrow(current_mut_tbl)
```

Add them into the supplementary table. And exclude the explained calls.
```{r}
supp_tbl <- bind_rows(supp_tbl, current_mut_tbl %>% select(-gdc_rowid))
gdc_tbl <- gdc_tbl %>% filter(!gdc_rowid %in% current_mut_tbl$gdc_rowid)
```

### Gene annotation change (Nonexonic in MC3 but exonic in GDC)
```{r}
current_mut_tbl <- gdc_tbl %>% 
    # The main effect in MC3 is non-exonic
    filter(
        mc3__NonExonic == 1,
        gdc_variant_classification %in% c('Missense_Mutation', 'Nonsense_Mutation', 'Silent', 'Nonstop_Mutation', "3'UTR", "5'UTR")
    ) %>%
    retreive_full_record_via_gdc() %>%
    mutate(
        concordance_status = 'recoverable GDC-unique',
        reason_not_included_in_public_gdc = NA_character_,
        reason_not_included_in_public_mc3 = 'filtered by NonExonic (gene annotation change)'
    )

nrow(current_mut_tbl)
```

Add them into the supplementary table. And exclude the explained calls.
```{r}
supp_tbl <- bind_rows(supp_tbl, current_mut_tbl %>% select(-gdc_rowid))
gdc_tbl <- gdc_tbl %>% filter(!gdc_rowid %in% current_mut_tbl$gdc_rowid)
```

### Nonexonic but unknown reason
The remaing GDC-unique calls were excluded in MC3 due to NonExonic, but we don't know why they made their way to the GDC public calls.

```{r}
current_mut_tbl <- gdc_tbl %>% 
    # The main effect in MC3 is non-exonic
    filter(mc3__NonExonic == 1) %>%
    retreive_full_record_via_gdc() %>%
    mutate(
        concordance_status = 'recoverable GDC-unique',
        reason_not_included_in_public_gdc = NA_character_,
        reason_not_included_in_public_mc3 = 'filtered by NonExonic (unknown)'
    )

nrow(current_mut_tbl)
```

Add them into the supplementary table. And exclude the explained calls.
```{r}
supp_tbl <- bind_rows(supp_tbl, current_mut_tbl %>% select(-gdc_rowid))
gdc_tbl <- gdc_tbl %>% filter(!gdc_rowid %in% current_mut_tbl$gdc_rowid)
```


### Panel of Normals (PoN) filter
The calls were excluded because they were also detected in the panel of normal samples. For those also detected in GDC's PoN, they were promoted due to validation sequencing.

```{r}
current_mut_tbl <- gdc_tbl %>% 
    filter(
        mc3__broad_PoN_v2 == 1,
        !(gdc__panel_of_normals == 1 | gdc_gdc__gdc_pon == 1)
    ) %>%
    retreive_full_record_via_gdc() %>%
    mutate(
        concordance_status = 'recoverable GDC-unique',
        reason_not_included_in_public_gdc = NA_character_,
        reason_not_included_in_public_mc3 = 'filtered by PoN (no PoN filter in GDC)'
    )

supp_tbl <- bind_rows(supp_tbl, current_mut_tbl %>% select(-gdc_rowid))
gdc_tbl <- gdc_tbl %>% filter(!gdc_rowid %in% current_mut_tbl$gdc_rowid)

current_mut_tbl <- gdc_tbl %>% 
    filter(
        mc3__broad_PoN_v2 == 1,
        gdc__panel_of_normals == 1 | gdc_gdc__gdc_pon == 1
    ) %>%
    retreive_full_record_via_gdc() %>%
    mutate(
        concordance_status = 'recoverable GDC-unique',
        reason_not_included_in_public_gdc = NA_character_,
        reason_not_included_in_public_mc3 = 'filtered by PoN (detected in validation sequencing)'
    )

supp_tbl <- bind_rows(supp_tbl, current_mut_tbl %>% select(-gdc_rowid))
gdc_tbl <- gdc_tbl %>% filter(!gdc_rowid %in% current_mut_tbl$gdc_rowid)
```

### Design regions of interest of the targeted sequencing
The remaining calls were excluded since they are out of the ROIs of the targeted sequencing. Most of them were included in GDC due to the validation sequencing.
```{r}
current_mut_tbl <- gdc_tbl %>% 
    filter(
        mc3__bitgt == 1,
        str_detect(gdc_validation_status, 'Valid')
    ) %>%
    retreive_full_record_via_gdc() %>%
    mutate(
        concordance_status = 'recoverable GDC-unique',
        reason_not_included_in_public_gdc = NA_character_,
        reason_not_included_in_public_mc3 = 'filtered by bitgt (detected in validation sequencing)'
    )

supp_tbl <- bind_rows(supp_tbl, current_mut_tbl %>% select(-gdc_rowid))
gdc_tbl <- gdc_tbl %>% filter(!gdc_rowid %in% current_mut_tbl$gdc_rowid)
```

```{r}
current_mut_tbl <- gdc_tbl %>% 
    filter(mc3__bitgt == 1) %>%
    retreive_full_record_via_gdc() %>%
    mutate(
        concordance_status = 'recoverable GDC-unique',
        reason_not_included_in_public_gdc = NA_character_,
        reason_not_included_in_public_mc3 = 'filtered by bitgt (unknown)'
    )

supp_tbl <- bind_rows(supp_tbl, current_mut_tbl %>% select(-gdc_rowid))
gdc_tbl <- gdc_tbl %>% filter(!gdc_rowid %in% current_mut_tbl$gdc_rowid)
```

And we left zero calls unexplained.
```{r}
nrow(gdc_tbl)
```



## Recoverable MC3-unique calls
We follow the order of the GDC filtering in their documentation.

We use the same idea to retrieve full record from `all_public_snv_tbl`. 
```{r}
retreive_full_record_via_mc3 <- function(mut_tbl) {
    mut_tbl %>%
        select(mc3_rowid) %>%
        left_join(all_public_snv_tbl, by = 'mc3_rowid') %>%
        select(
            mc3_rowid,
            chromosome, start_position, end_position, tumor_sample_barcode, cancer_type,
            reference_allele, tumor_seq_allele1=mc3_tumor_seq_allele1, tumor_seq_allele2,
            hgvsc=mc3_hgvsc, hgvsp=mc3_hgvsp, 
            variant_classification=mc3_variant_classification, variant_type=mc3_variant_type,
            hugo_symbol=mc3_hugo_symbol, transcript_id=mc3_transcript_id
        )
}
```


### GDC filtering stage 1 (multiallelic and nonselectedaliquot)
```{r}
current_mut_tbl <- mc3_tbl %>% 
    filter(
        gdc_gdc__nonselectedaliquot == 1 | gdc_gdc__multiallelic == 1
    ) %>%
    retreive_full_record_via_mc3() %>%
    mutate(
        concordance_status = 'recoverable MC3-unique',
        reason_not_included_in_public_gdc = 'filtering stage 1 (multi-alleleic or non-selected aliquot)',
        reason_not_included_in_public_mc3 = NA_character_
    )

current_mut_tbl %>% head()
nrow(current_mut_tbl)
```

Double check the full record retrieval:
```{r}
mc3_tbl %>% 
    filter(gdc_gdc__nonselectedaliquot == 1 | gdc_gdc__multiallelic == 1) %>%
    select(
        mc3_rowid,
        chromosome, start_position, end_position, tumor_sample_barcode, cancer_type,
        reference_allele, tumor_seq_allele1=mc3_tumor_seq_allele1, tumor_seq_allele2
    ) %>%
    head()
```

Add the expalined calls into the supplementary table. And exclude the explained calls from the remaining recoverable MC3-unique calls.
```{r}
supp_tbl <- bind_rows(supp_tbl, current_mut_tbl %>% select(-mc3_rowid))
mc3_tbl <- mc3_tbl %>% filter(!mc3_rowid %in% current_mut_tbl$mc3_rowid)
```


### GDC filtering stage 2 (valid somatic)
None of the calls were detected in the validation sequencing.
```{r}
mc3_tbl %>% filter(str_detect(gdc_validation_status, 'Valid'))
```


### GDC filtering stage 3 (`gdc_filter`)
Mutation calls were excluded if they have additional filters reported by the mutation callers.
```{r}
current_mut_tbl <- mc3_tbl %>% 
    filter(
        !(gdc_filter %in% c(
            'PASS', 'panel_of_normals', 
            'PASS;panel_of_normals', 'panel_of_normals;PASS'
        ))
    ) %>%
    # We add the filters into the explanation
    mutate(
        explanation=as.character(glue::glue('filtering stage 3 (additional filters: {gdc_filter})'))
    ) %>%
    select(mc3_rowid, explanation) %>%
    left_join(all_public_snv_tbl, by = 'mc3_rowid') %>%
    select(
        mc3_rowid,
        chromosome, start_position, end_position, tumor_sample_barcode, cancer_type,
        reference_allele, tumor_seq_allele1=mc3_tumor_seq_allele1, tumor_seq_allele2,
        hgvsc=mc3_hgvsc, hgvsp=mc3_hgvsp, 
        variant_classification=mc3_variant_classification, variant_type=mc3_variant_type,
        hugo_symbol=mc3_hugo_symbol, transcript_id=mc3_transcript_id, explanation
    ) %>%
    mutate(
        concordance_status = 'recoverable MC3-unique',
        reason_not_included_in_public_gdc = explanation,
        reason_not_included_in_public_mc3 = NA_character_
    ) %>%
    select(-explanation)

nrow(current_mut_tbl)
head(current_mut_tbl$reason_not_included_in_public_gdc)
```

```{r}
supp_tbl <- bind_rows(supp_tbl, current_mut_tbl %>% select(-mc3_rowid))
mc3_tbl <- mc3_tbl %>% filter(!mc3_rowid %in% current_mut_tbl$mc3_rowid)
```


### GDC filtering stage 5 (`gdc_gdc_filter`)
GDC added additional filters to the calls and the calls with the following GDC filters were excluded: 

- ndp
- NonExonic
- bitgt
- gdc_pon

```{r}
current_mut_tbl <- mc3_tbl %>% 
    filter(
        gdc_gdc__ndp == 1 | gdc_gdc__NonExonic == 1 | gdc_gdc__bitgt == 1 | gdc_gdc__gdc_pon == 1
    ) %>%
    mutate(
        # Collapase the multiple indicators into one column
        stage5_filters = pmap(
            .,
            function(gdc_gdc__ndp, gdc_gdc__NonExonic, gdc_gdc__bitgt, gdc_gdc__gdc_pon, ...) {
                str_c(
                    subset(
                        c('ndp', 'NonExonic', 'bitgt', 'gdc_pon'), 
                        as.logical(c(gdc_gdc__ndp, gdc_gdc__NonExonic, gdc_gdc__bitgt, gdc_gdc__gdc_pon))
                    ), 
                    collapse = ';'
                )
            }
        )
    ) %>%
    mutate(
        explanation=as.character(glue::glue('filtering stage 5 (additional filters: {stage5_filters})'))
    ) %>%
    select(mc3_rowid, explanation) %>%
    left_join(all_public_snv_tbl, by = 'mc3_rowid') %>%
    select(
        mc3_rowid,
        chromosome, start_position, end_position, tumor_sample_barcode, cancer_type,
        reference_allele, tumor_seq_allele1=mc3_tumor_seq_allele1, tumor_seq_allele2,
        hgvsc=mc3_hgvsc, hgvsp=mc3_hgvsp, 
        variant_classification=mc3_variant_classification, variant_type=mc3_variant_type,
        hugo_symbol=mc3_hugo_symbol, transcript_id=mc3_transcript_id, explanation
    ) %>%
    mutate(
        concordance_status = 'recoverable MC3-unique',
        reason_not_included_in_public_gdc = explanation,
        reason_not_included_in_public_mc3 = NA_character_
    ) %>%
    select(-explanation)

nrow(current_mut_tbl)
```

```{r}
supp_tbl <- bind_rows(supp_tbl, current_mut_tbl %>% select(-mc3_rowid))
mc3_tbl <- mc3_tbl %>% filter(!mc3_rowid %in% current_mut_tbl$mc3_rowid)
```

### GDC filtering stage 7 (dbSNP)
```{r}
current_mut_tbl <- mc3_tbl %>% 
    filter(
        str_detect(gdc_dbsnp_rs, '^rs')
    ) %>%
    retreive_full_record_via_mc3() %>%
    mutate(
        concordance_status = 'recoverable MC3-unique',
        reason_not_included_in_public_gdc = 'filtering stage 7 (known SNP in dbSNP)',
        reason_not_included_in_public_mc3 = NA_character_
    )

current_mut_tbl %>% head()
nrow(current_mut_tbl)
```

```{r}
supp_tbl <- bind_rows(supp_tbl, current_mut_tbl %>% select(-mc3_rowid))
mc3_tbl <- mc3_tbl %>% filter(!mc3_rowid %in% current_mut_tbl$mc3_rowid)
```


We left zero mutation unexplained.
```{r}
nrow(mc3_tbl)
```


## Export supplementary table
```{r}
supp_tbl <- supp_tbl %>%
    mutate(
        chromosome = factor(chromosome, c(str_c('chr', seq(1, 22)), 'chrX', 'chrY', 'chrM')),
        hgvsc = str_replace(replace_na(hgvsc, ''), '^\\.$', ''),
        hgvsp = str_replace(replace_na(hgvsp, ''), '^\\.$', '')
    ) %>%
    arrange(chromosome, start_position, -end_position, cancer_type, tumor_sample_barcode)

write_tsv(supp_tbl, '../processed_data/supplementary_table.tsv.gz')
```

