{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sqlite3\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<sqlite3.Cursor at 0x1062e57a0>"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "conn = sqlite3.connect('../processed_data/all_variants.sqlite')\n",
    "conn.executescript('''\\\n",
    "PRAGMA cache_size=-4192000;\n",
    "PRAGMA temp_store=MEMORY;\n",
    "PRAGMA journal_mode=OFF;\n",
    "''')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Filter only the samples shared by GDC and MC3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MC3: 1902, GDC: 1965 samples\n"
     ]
    }
   ],
   "source": [
    "c = conn.execute('''SELECT DISTINCT tumor_sample_barcode FROM mc3_selected''')\n",
    "mc3_samples = [t[0] for t in c.fetchall()]\n",
    "c = conn.execute('''SELECT DISTINCT tumor_sample_barcode FROM gdc_grouped_callers''')\n",
    "gdc_samples = [t[0] for t in c.fetchall()]\n",
    "print(f'MC3: {len(mc3_samples)}, GDC: {len(gdc_samples)} samples')\n",
    "\n",
    "gdc_only_samples = set(gdc_samples) - set(mc3_samples)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "_gdc_samples_sql_literal = ','.join(f\"'{sample}'\" for sample in gdc_only_samples)\n",
    "c = conn.execute(f'''\\\n",
    "CREATE VIEW IF NOT EXISTS gdc_grp_shared_samples AS\n",
    "SELECT * FROM gdc_grouped_callers\n",
    "WHERE tumor_sample_barcode NOT IN ({_gdc_samples_sql_literal})\n",
    "''')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_variant_class = pd.merge(\n",
    "    pd.read_sql('''\\\n",
    "SELECT variant_classification, count(*) AS 'mc3_count'\n",
    "FROM mc3_selected\n",
    "GROUP BY variant_classification\n",
    "ORDER BY mc3_count DESC\n",
    "''', conn),\n",
    "    pd.read_sql('''\\\n",
    "SELECT variant_classification, count(*) AS 'gdc_count'\n",
    "FROM gdc_grp_shared_samples\n",
    "GROUP BY variant_classification\n",
    "ORDER BY gdc_count DESC\n",
    "''', conn),\n",
    "    how='outer',\n",
    "    on='variant_classification'\n",
    ")\n",
    "\n",
    "df_variant_class = df_variant_class.fillna(0).assign(\n",
    "    mc3_count = lambda df: df.mc3_count.astype(int),\n",
    "    gdc_count = lambda df: df.gdc_count.astype(int)\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant_classification</th>\n",
       "      <th>mc3_count</th>\n",
       "      <th>gdc_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>236394</td>\n",
       "      <td>249236</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Silent</td>\n",
       "      <td>91918</td>\n",
       "      <td>91839</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3'UTR</td>\n",
       "      <td>31933</td>\n",
       "      <td>40730</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Frame_Shift_Del</td>\n",
       "      <td>24386</td>\n",
       "      <td>24221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Nonsense_Mutation</td>\n",
       "      <td>18048</td>\n",
       "      <td>20366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Intron</td>\n",
       "      <td>14440</td>\n",
       "      <td>25368</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>5'UTR</td>\n",
       "      <td>10959</td>\n",
       "      <td>12096</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Splice_Site</td>\n",
       "      <td>7080</td>\n",
       "      <td>6109</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Frame_Shift_Ins</td>\n",
       "      <td>6726</td>\n",
       "      <td>11455</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>RNA</td>\n",
       "      <td>5301</td>\n",
       "      <td>8391</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>In_Frame_Del</td>\n",
       "      <td>2297</td>\n",
       "      <td>2746</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>3'Flank</td>\n",
       "      <td>2032</td>\n",
       "      <td>3702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>5'Flank</td>\n",
       "      <td>1874</td>\n",
       "      <td>3190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Translation_Start_Site</td>\n",
       "      <td>315</td>\n",
       "      <td>346</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Nonstop_Mutation</td>\n",
       "      <td>260</td>\n",
       "      <td>276</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>In_Frame_Ins</td>\n",
       "      <td>165</td>\n",
       "      <td>1920</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Splice_Region</td>\n",
       "      <td>0</td>\n",
       "      <td>5016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>IGR</td>\n",
       "      <td>0</td>\n",
       "      <td>49</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    variant_classification  mc3_count  gdc_count\n",
       "0        Missense_Mutation     236394     249236\n",
       "1                   Silent      91918      91839\n",
       "2                    3'UTR      31933      40730\n",
       "3          Frame_Shift_Del      24386      24221\n",
       "4        Nonsense_Mutation      18048      20366\n",
       "5                   Intron      14440      25368\n",
       "6                    5'UTR      10959      12096\n",
       "7              Splice_Site       7080       6109\n",
       "8          Frame_Shift_Ins       6726      11455\n",
       "9                      RNA       5301       8391\n",
       "10            In_Frame_Del       2297       2746\n",
       "11                 3'Flank       2032       3702\n",
       "12                 5'Flank       1874       3190\n",
       "13  Translation_Start_Site        315        346\n",
       "14        Nonstop_Mutation        260        276\n",
       "15            In_Frame_Ins        165       1920\n",
       "16           Splice_Region          0       5016\n",
       "17                     IGR          0         49"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_variant_class"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "variant_classification    Missense_MutationSilent3'UTRFrame_Shift_DelNon...\n",
       "mc3_count                                                            454128\n",
       "gdc_count                                                            507056\n",
       "dtype: object"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_variant_class.sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_per_tx_sample_variant_type = pd.read_sql('''\\\n",
    "WITH gdc_by_tx AS (\n",
    "    SELECT hugo_symbol, transcript_id, variant_classification, tumor_sample_barcode, count(*) AS 'gdc_count'\n",
    "    FROM gdc_grp_shared_samples\n",
    "    GROUP BY hugo_symbol, transcript_id, variant_classification, tumor_sample_barcode\n",
    "), mc3_by_tx AS (\n",
    "    SELECT hugo_symbol, transcript_id, variant_classification, tumor_sample_barcode, count(*) AS 'mc3_count'\n",
    "    FROM mc3_selected\n",
    "    GROUP BY hugo_symbol, transcript_id, variant_classification, tumor_sample_barcode\n",
    ")\n",
    "SELECT hugo_symbol, transcript_id, variant_classification, tumor_sample_barcode, gdc_count, mc3_count FROM gdc_by_tx\n",
    "LEFT JOIN mc3_by_tx \n",
    "USING (hugo_symbol, transcript_id, variant_classification, tumor_sample_barcode)\n",
    "UNION ALL\n",
    "SELECT hugo_symbol, transcript_id, variant_classification, tumor_sample_barcode, gdc_count, mc3_count FROM mc3_by_tx\n",
    "LEFT JOIN gdc_by_tx\n",
    "USING (hugo_symbol, transcript_id, variant_classification, tumor_sample_barcode)\n",
    "WHERE gdc_by_tx.gdc_count IS NULL\n",
    "''', conn)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_per_tx_sample_variant_type = (\n",
    "    df_per_tx_sample_variant_type\n",
    "    .fillna(0)\n",
    "    .assign(\n",
    "        gdc_count=lambda df: df.gdc_count.astype(int),\n",
    "        mc3_count=lambda df: df.mc3_count.astype(int))\n",
    "    .assign(\n",
    "        diff=lambda df: df.gdc_count-df.mc3_count\n",
    "    )\n",
    "    .sort_values(['diff'], ascending=False)\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hugo_symbol</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>variant_classification</th>\n",
       "      <th>tumor_sample_barcode</th>\n",
       "      <th>gdc_count</th>\n",
       "      <th>mc3_count</th>\n",
       "      <th>diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>430735</th>\n",
       "      <td>TTN</td>\n",
       "      <td>ENST00000591111</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>TCGA-AA-A010-01A-01D-A17O-10</td>\n",
       "      <td>29</td>\n",
       "      <td>0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>430863</th>\n",
       "      <td>TTN</td>\n",
       "      <td>ENST00000591111</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>TCGA-CA-6717-01A-11D-1835-10</td>\n",
       "      <td>29</td>\n",
       "      <td>0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>430776</th>\n",
       "      <td>TTN</td>\n",
       "      <td>ENST00000591111</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>TCGA-AN-A046-01A-21W-A050-09</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>430734</th>\n",
       "      <td>TTN</td>\n",
       "      <td>ENST00000591111</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>TCGA-AA-A00N-01A-02D-A17O-10</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>430813</th>\n",
       "      <td>TTN</td>\n",
       "      <td>ENST00000591111</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>TCGA-AZ-4315-01A-01D-1408-10</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       hugo_symbol    transcript_id variant_classification  \\\n",
       "430735         TTN  ENST00000591111      Missense_Mutation   \n",
       "430863         TTN  ENST00000591111      Missense_Mutation   \n",
       "430776         TTN  ENST00000591111      Missense_Mutation   \n",
       "430734         TTN  ENST00000591111      Missense_Mutation   \n",
       "430813         TTN  ENST00000591111      Missense_Mutation   \n",
       "\n",
       "                tumor_sample_barcode  gdc_count  mc3_count  diff  \n",
       "430735  TCGA-AA-A010-01A-01D-A17O-10         29          0    29  \n",
       "430863  TCGA-CA-6717-01A-11D-1835-10         29          0    29  \n",
       "430776  TCGA-AN-A046-01A-21W-A050-09         28          0    28  \n",
       "430734  TCGA-AA-A00N-01A-02D-A17O-10         28          0    28  \n",
       "430813  TCGA-AZ-4315-01A-01D-1408-10         23          0    23  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_per_tx_sample_variant_type.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hugo_symbol</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>variant_classification</th>\n",
       "      <th>tumor_sample_barcode</th>\n",
       "      <th>gdc_count</th>\n",
       "      <th>mc3_count</th>\n",
       "      <th>diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>420141</th>\n",
       "      <td>TP53</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Splice_Region</td>\n",
       "      <td>TCGA-13-0755-01A-01W-0371-08</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>588258</th>\n",
       "      <td>TP53</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Silent</td>\n",
       "      <td>TCGA-13-0755-01A-01W-0371-08</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       hugo_symbol    transcript_id variant_classification  \\\n",
       "420141        TP53  ENST00000269305          Splice_Region   \n",
       "588258        TP53  ENST00000269305                 Silent   \n",
       "\n",
       "                tumor_sample_barcode  gdc_count  mc3_count  diff  \n",
       "420141  TCGA-13-0755-01A-01W-0371-08          2          0     2  \n",
       "588258  TCGA-13-0755-01A-01W-0371-08          0          1    -1  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_per_tx_sample_variant_type[\n",
    "    (df_per_tx_sample_variant_type['hugo_symbol'] == 'TP53') &\n",
    "    (df_per_tx_sample_variant_type['tumor_sample_barcode'] == 'TCGA-13-0755-01A-01W-0371-08')\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_per_tx_sample = (\n",
    "    df_per_tx_sample_variant_type\n",
    "    .groupby(['hugo_symbol', 'transcript_id', 'tumor_sample_barcode'])\n",
    "    .sum()\n",
    "    .sort_values(['diff'], ascending=False)\n",
    "    .reset_index()\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hugo_symbol</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>tumor_sample_barcode</th>\n",
       "      <th>gdc_count</th>\n",
       "      <th>mc3_count</th>\n",
       "      <th>diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TTN</td>\n",
       "      <td>ENST00000591111</td>\n",
       "      <td>TCGA-CA-6717-01A-11D-1835-10</td>\n",
       "      <td>68</td>\n",
       "      <td>0</td>\n",
       "      <td>68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TTN</td>\n",
       "      <td>ENST00000591111</td>\n",
       "      <td>TCGA-AA-A010-01A-01D-A17O-10</td>\n",
       "      <td>48</td>\n",
       "      <td>0</td>\n",
       "      <td>48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TTN</td>\n",
       "      <td>ENST00000591111</td>\n",
       "      <td>TCGA-AN-A046-01A-21W-A050-09</td>\n",
       "      <td>45</td>\n",
       "      <td>0</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TTN</td>\n",
       "      <td>ENST00000591111</td>\n",
       "      <td>TCGA-AZ-4315-01A-01D-1408-10</td>\n",
       "      <td>44</td>\n",
       "      <td>0</td>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TTN</td>\n",
       "      <td>ENST00000591111</td>\n",
       "      <td>TCGA-AA-A00N-01A-02D-A17O-10</td>\n",
       "      <td>40</td>\n",
       "      <td>0</td>\n",
       "      <td>40</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  hugo_symbol    transcript_id          tumor_sample_barcode  gdc_count  \\\n",
       "0         TTN  ENST00000591111  TCGA-CA-6717-01A-11D-1835-10         68   \n",
       "1         TTN  ENST00000591111  TCGA-AA-A010-01A-01D-A17O-10         48   \n",
       "2         TTN  ENST00000591111  TCGA-AN-A046-01A-21W-A050-09         45   \n",
       "3         TTN  ENST00000591111  TCGA-AZ-4315-01A-01D-1408-10         44   \n",
       "4         TTN  ENST00000591111  TCGA-AA-A00N-01A-02D-A17O-10         40   \n",
       "\n",
       "   mc3_count  diff  \n",
       "0          0    68  \n",
       "1          0    48  \n",
       "2          0    45  \n",
       "3          0    44  \n",
       "4          0    40  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_per_tx_sample.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_per_tx = (\n",
    "    df_per_tx_sample.groupby(['hugo_symbol', 'transcript_id'])\n",
    "    .sum()\n",
    "    .reset_index()\n",
    "    .assign(absdiff = lambda df: df['diff'].abs())\n",
    "    .sort_values(['absdiff'], ascending=False)\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hugo_symbol</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gdc_count</th>\n",
       "      <th>mc3_count</th>\n",
       "      <th>diff</th>\n",
       "      <th>absdiff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>673</th>\n",
       "      <td>ADGRV1</td>\n",
       "      <td>ENST00000405460</td>\n",
       "      <td>217</td>\n",
       "      <td>0</td>\n",
       "      <td>217</td>\n",
       "      <td>217</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1026</th>\n",
       "      <td>ALMS1</td>\n",
       "      <td>ENST00000613296</td>\n",
       "      <td>155</td>\n",
       "      <td>0</td>\n",
       "      <td>155</td>\n",
       "      <td>155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1389</th>\n",
       "      <td>APC</td>\n",
       "      <td>ENST00000257430</td>\n",
       "      <td>598</td>\n",
       "      <td>0</td>\n",
       "      <td>598</td>\n",
       "      <td>598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1390</th>\n",
       "      <td>APC</td>\n",
       "      <td>ENST00000457016</td>\n",
       "      <td>0</td>\n",
       "      <td>559</td>\n",
       "      <td>-559</td>\n",
       "      <td>559</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3587</th>\n",
       "      <td>CCDC168</td>\n",
       "      <td>ENST00000322527</td>\n",
       "      <td>284</td>\n",
       "      <td>93</td>\n",
       "      <td>191</td>\n",
       "      <td>191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5184</th>\n",
       "      <td>CSMD1</td>\n",
       "      <td>ENST00000520002</td>\n",
       "      <td>257</td>\n",
       "      <td>0</td>\n",
       "      <td>257</td>\n",
       "      <td>257</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5185</th>\n",
       "      <td>CSMD1</td>\n",
       "      <td>ENST00000537824</td>\n",
       "      <td>0</td>\n",
       "      <td>265</td>\n",
       "      <td>-265</td>\n",
       "      <td>265</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5759</th>\n",
       "      <td>DCHS2</td>\n",
       "      <td>ENST00000623607</td>\n",
       "      <td>164</td>\n",
       "      <td>0</td>\n",
       "      <td>164</td>\n",
       "      <td>164</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5757</th>\n",
       "      <td>DCHS2</td>\n",
       "      <td>ENST00000357232</td>\n",
       "      <td>0</td>\n",
       "      <td>155</td>\n",
       "      <td>-155</td>\n",
       "      <td>155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6164</th>\n",
       "      <td>DNAH11</td>\n",
       "      <td>ENST00000409508</td>\n",
       "      <td>248</td>\n",
       "      <td>0</td>\n",
       "      <td>248</td>\n",
       "      <td>248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6163</th>\n",
       "      <td>DNAH11</td>\n",
       "      <td>ENST00000328843</td>\n",
       "      <td>0</td>\n",
       "      <td>230</td>\n",
       "      <td>-230</td>\n",
       "      <td>230</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6176</th>\n",
       "      <td>DNAH2</td>\n",
       "      <td>ENST00000389173</td>\n",
       "      <td>170</td>\n",
       "      <td>0</td>\n",
       "      <td>170</td>\n",
       "      <td>170</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6177</th>\n",
       "      <td>DNAH2</td>\n",
       "      <td>ENST00000572933</td>\n",
       "      <td>0</td>\n",
       "      <td>169</td>\n",
       "      <td>-169</td>\n",
       "      <td>169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6423</th>\n",
       "      <td>DST</td>\n",
       "      <td>ENST00000312431</td>\n",
       "      <td>381</td>\n",
       "      <td>0</td>\n",
       "      <td>381</td>\n",
       "      <td>381</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6422</th>\n",
       "      <td>DST</td>\n",
       "      <td>ENST00000244364</td>\n",
       "      <td>0</td>\n",
       "      <td>329</td>\n",
       "      <td>-329</td>\n",
       "      <td>329</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7186</th>\n",
       "      <td>EYS</td>\n",
       "      <td>ENST00000370616</td>\n",
       "      <td>163</td>\n",
       "      <td>0</td>\n",
       "      <td>163</td>\n",
       "      <td>163</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7671</th>\n",
       "      <td>FAT3</td>\n",
       "      <td>ENST00000525166</td>\n",
       "      <td>281</td>\n",
       "      <td>0</td>\n",
       "      <td>281</td>\n",
       "      <td>281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7670</th>\n",
       "      <td>FAT3</td>\n",
       "      <td>ENST00000298047</td>\n",
       "      <td>0</td>\n",
       "      <td>258</td>\n",
       "      <td>-258</td>\n",
       "      <td>258</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7781</th>\n",
       "      <td>FCGBP</td>\n",
       "      <td>ENST00000616721</td>\n",
       "      <td>197</td>\n",
       "      <td>0</td>\n",
       "      <td>197</td>\n",
       "      <td>197</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7780</th>\n",
       "      <td>FCGBP</td>\n",
       "      <td>ENST00000221347</td>\n",
       "      <td>0</td>\n",
       "      <td>156</td>\n",
       "      <td>-156</td>\n",
       "      <td>156</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8422</th>\n",
       "      <td>GATA3</td>\n",
       "      <td>ENST00000346208</td>\n",
       "      <td>183</td>\n",
       "      <td>0</td>\n",
       "      <td>183</td>\n",
       "      <td>183</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8423</th>\n",
       "      <td>GATA3</td>\n",
       "      <td>ENST00000379328</td>\n",
       "      <td>0</td>\n",
       "      <td>175</td>\n",
       "      <td>-175</td>\n",
       "      <td>175</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9048</th>\n",
       "      <td>GPR98</td>\n",
       "      <td>ENST00000405460</td>\n",
       "      <td>0</td>\n",
       "      <td>206</td>\n",
       "      <td>-206</td>\n",
       "      <td>206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10073</th>\n",
       "      <td>HUWE1</td>\n",
       "      <td>ENST00000262854</td>\n",
       "      <td>159</td>\n",
       "      <td>0</td>\n",
       "      <td>159</td>\n",
       "      <td>159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12586</th>\n",
       "      <td>MACF1</td>\n",
       "      <td>ENST00000372915</td>\n",
       "      <td>275</td>\n",
       "      <td>0</td>\n",
       "      <td>275</td>\n",
       "      <td>275</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12587</th>\n",
       "      <td>MACF1</td>\n",
       "      <td>ENST00000545844</td>\n",
       "      <td>0</td>\n",
       "      <td>201</td>\n",
       "      <td>-201</td>\n",
       "      <td>201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13965</th>\n",
       "      <td>MUC2</td>\n",
       "      <td>ENST00000361558</td>\n",
       "      <td>188</td>\n",
       "      <td>0</td>\n",
       "      <td>188</td>\n",
       "      <td>188</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13971</th>\n",
       "      <td>MUC4</td>\n",
       "      <td>ENST00000463781</td>\n",
       "      <td>474</td>\n",
       "      <td>148</td>\n",
       "      <td>326</td>\n",
       "      <td>326</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14023</th>\n",
       "      <td>MYCBP2</td>\n",
       "      <td>ENST00000357337</td>\n",
       "      <td>167</td>\n",
       "      <td>0</td>\n",
       "      <td>167</td>\n",
       "      <td>167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14024</th>\n",
       "      <td>MYCBP2</td>\n",
       "      <td>ENST00000544440</td>\n",
       "      <td>0</td>\n",
       "      <td>159</td>\n",
       "      <td>-159</td>\n",
       "      <td>159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14436</th>\n",
       "      <td>NEB</td>\n",
       "      <td>ENST00000172853</td>\n",
       "      <td>304</td>\n",
       "      <td>0</td>\n",
       "      <td>304</td>\n",
       "      <td>304</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14437</th>\n",
       "      <td>NEB</td>\n",
       "      <td>ENST00000397345</td>\n",
       "      <td>0</td>\n",
       "      <td>299</td>\n",
       "      <td>-299</td>\n",
       "      <td>299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15159</th>\n",
       "      <td>OBSCN</td>\n",
       "      <td>ENST00000422127</td>\n",
       "      <td>380</td>\n",
       "      <td>0</td>\n",
       "      <td>380</td>\n",
       "      <td>380</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15160</th>\n",
       "      <td>OBSCN</td>\n",
       "      <td>ENST00000570156</td>\n",
       "      <td>0</td>\n",
       "      <td>391</td>\n",
       "      <td>-391</td>\n",
       "      <td>391</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16213</th>\n",
       "      <td>PDE4DIP</td>\n",
       "      <td>ENST00000369356</td>\n",
       "      <td>0</td>\n",
       "      <td>175</td>\n",
       "      <td>-175</td>\n",
       "      <td>175</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19788</th>\n",
       "      <td>SACS</td>\n",
       "      <td>ENST00000382292</td>\n",
       "      <td>198</td>\n",
       "      <td>0</td>\n",
       "      <td>198</td>\n",
       "      <td>198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19789</th>\n",
       "      <td>SACS</td>\n",
       "      <td>ENST00000382298</td>\n",
       "      <td>0</td>\n",
       "      <td>180</td>\n",
       "      <td>-180</td>\n",
       "      <td>180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22202</th>\n",
       "      <td>SYNE2</td>\n",
       "      <td>ENST00000344113</td>\n",
       "      <td>259</td>\n",
       "      <td>0</td>\n",
       "      <td>259</td>\n",
       "      <td>259</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22203</th>\n",
       "      <td>SYNE2</td>\n",
       "      <td>ENST00000358025</td>\n",
       "      <td>5</td>\n",
       "      <td>235</td>\n",
       "      <td>-230</td>\n",
       "      <td>230</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22628</th>\n",
       "      <td>TENM1</td>\n",
       "      <td>ENST00000371130</td>\n",
       "      <td>166</td>\n",
       "      <td>0</td>\n",
       "      <td>166</td>\n",
       "      <td>166</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22650</th>\n",
       "      <td>TET2</td>\n",
       "      <td>ENST00000380013</td>\n",
       "      <td>158</td>\n",
       "      <td>0</td>\n",
       "      <td>158</td>\n",
       "      <td>158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23462</th>\n",
       "      <td>TP53</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>1260</td>\n",
       "      <td>1003</td>\n",
       "      <td>257</td>\n",
       "      <td>257</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24078</th>\n",
       "      <td>TTN</td>\n",
       "      <td>ENST00000591111</td>\n",
       "      <td>1886</td>\n",
       "      <td>0</td>\n",
       "      <td>1886</td>\n",
       "      <td>1886</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24077</th>\n",
       "      <td>TTN</td>\n",
       "      <td>ENST00000589042</td>\n",
       "      <td>0</td>\n",
       "      <td>1694</td>\n",
       "      <td>-1694</td>\n",
       "      <td>1694</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25035</th>\n",
       "      <td>XIRP2</td>\n",
       "      <td>ENST00000628543</td>\n",
       "      <td>215</td>\n",
       "      <td>0</td>\n",
       "      <td>215</td>\n",
       "      <td>215</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25034</th>\n",
       "      <td>XIRP2</td>\n",
       "      <td>ENST00000409195</td>\n",
       "      <td>13</td>\n",
       "      <td>214</td>\n",
       "      <td>-201</td>\n",
       "      <td>201</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      hugo_symbol    transcript_id  gdc_count  mc3_count  diff  absdiff\n",
       "673        ADGRV1  ENST00000405460        217          0   217      217\n",
       "1026        ALMS1  ENST00000613296        155          0   155      155\n",
       "1389          APC  ENST00000257430        598          0   598      598\n",
       "1390          APC  ENST00000457016          0        559  -559      559\n",
       "3587      CCDC168  ENST00000322527        284         93   191      191\n",
       "5184        CSMD1  ENST00000520002        257          0   257      257\n",
       "5185        CSMD1  ENST00000537824          0        265  -265      265\n",
       "5759        DCHS2  ENST00000623607        164          0   164      164\n",
       "5757        DCHS2  ENST00000357232          0        155  -155      155\n",
       "6164       DNAH11  ENST00000409508        248          0   248      248\n",
       "6163       DNAH11  ENST00000328843          0        230  -230      230\n",
       "6176        DNAH2  ENST00000389173        170          0   170      170\n",
       "6177        DNAH2  ENST00000572933          0        169  -169      169\n",
       "6423          DST  ENST00000312431        381          0   381      381\n",
       "6422          DST  ENST00000244364          0        329  -329      329\n",
       "7186          EYS  ENST00000370616        163          0   163      163\n",
       "7671         FAT3  ENST00000525166        281          0   281      281\n",
       "7670         FAT3  ENST00000298047          0        258  -258      258\n",
       "7781        FCGBP  ENST00000616721        197          0   197      197\n",
       "7780        FCGBP  ENST00000221347          0        156  -156      156\n",
       "8422        GATA3  ENST00000346208        183          0   183      183\n",
       "8423        GATA3  ENST00000379328          0        175  -175      175\n",
       "9048        GPR98  ENST00000405460          0        206  -206      206\n",
       "10073       HUWE1  ENST00000262854        159          0   159      159\n",
       "12586       MACF1  ENST00000372915        275          0   275      275\n",
       "12587       MACF1  ENST00000545844          0        201  -201      201\n",
       "13965        MUC2  ENST00000361558        188          0   188      188\n",
       "13971        MUC4  ENST00000463781        474        148   326      326\n",
       "14023      MYCBP2  ENST00000357337        167          0   167      167\n",
       "14024      MYCBP2  ENST00000544440          0        159  -159      159\n",
       "14436         NEB  ENST00000172853        304          0   304      304\n",
       "14437         NEB  ENST00000397345          0        299  -299      299\n",
       "15159       OBSCN  ENST00000422127        380          0   380      380\n",
       "15160       OBSCN  ENST00000570156          0        391  -391      391\n",
       "16213     PDE4DIP  ENST00000369356          0        175  -175      175\n",
       "19788        SACS  ENST00000382292        198          0   198      198\n",
       "19789        SACS  ENST00000382298          0        180  -180      180\n",
       "22202       SYNE2  ENST00000344113        259          0   259      259\n",
       "22203       SYNE2  ENST00000358025          5        235  -230      230\n",
       "22628       TENM1  ENST00000371130        166          0   166      166\n",
       "22650        TET2  ENST00000380013        158          0   158      158\n",
       "23462        TP53  ENST00000269305       1260       1003   257      257\n",
       "24078         TTN  ENST00000591111       1886          0  1886     1886\n",
       "24077         TTN  ENST00000589042          0       1694 -1694     1694\n",
       "25035       XIRP2  ENST00000628543        215          0   215      215\n",
       "25034       XIRP2  ENST00000409195         13        214  -201      201"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_per_tx[df_per_tx['absdiff'] >= 150].sort_values(['hugo_symbol', 'mc3_count'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Export data to CSV\n",
    "df_per_tx.to_csv('../processed_data/variant_count_diff.per_tx.csv', index=False)\n",
    "df_per_tx_sample.to_csv('../processed_data/variant_count_diff.per_tx_sample.csv', index=False)\n",
    "df_per_tx_sample_variant_type.to_csv('../processed_data/variant_count_diff.per_tx_sample_type.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hugo_symbol</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gdc_count</th>\n",
       "      <th>mc3_count</th>\n",
       "      <th>diff</th>\n",
       "      <th>absdiff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>24201</th>\n",
       "      <td>U2AF1L4</td>\n",
       "      <td>ENST00000412391</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24199</th>\n",
       "      <td>U2AF1</td>\n",
       "      <td>ENST00000291552</td>\n",
       "      <td>4</td>\n",
       "      <td>18</td>\n",
       "      <td>-14</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24200</th>\n",
       "      <td>U2AF1L4</td>\n",
       "      <td>ENST00000292879</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>-9</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      hugo_symbol    transcript_id  gdc_count  mc3_count  diff  absdiff\n",
       "24201     U2AF1L4  ENST00000412391         16          0    16       16\n",
       "24199       U2AF1  ENST00000291552          4         18   -14       14\n",
       "24200     U2AF1L4  ENST00000292879          0          9    -9        9"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_per_tx[df_per_tx['hugo_symbol'].str.startswith('U2AF1')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hugo_symbol</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>tumor_sample_barcode</th>\n",
       "      <th>gdc_count</th>\n",
       "      <th>mc3_count</th>\n",
       "      <th>diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>APC</td>\n",
       "      <td>ENST00000257430</td>\n",
       "      <td>TCGA-CA-6717-01A-11D-1835-10</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>APC</td>\n",
       "      <td>ENST00000257430</td>\n",
       "      <td>TCGA-AN-A046-01A-21W-A050-09</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>APC</td>\n",
       "      <td>ENST00000257430</td>\n",
       "      <td>TCGA-AA-A010-01A-01D-A17O-10</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>APC</td>\n",
       "      <td>ENST00000257430</td>\n",
       "      <td>TCGA-CM-6166-01A-11D-1650-10</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>311</th>\n",
       "      <td>APC</td>\n",
       "      <td>ENST00000257430</td>\n",
       "      <td>TCGA-AA-3947-01A-01D-1981-10</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    hugo_symbol    transcript_id          tumor_sample_barcode  gdc_count  \\\n",
       "73          APC  ENST00000257430  TCGA-CA-6717-01A-11D-1835-10          9   \n",
       "130         APC  ENST00000257430  TCGA-AN-A046-01A-21W-A050-09          7   \n",
       "131         APC  ENST00000257430  TCGA-AA-A010-01A-01D-A17O-10          7   \n",
       "145         APC  ENST00000257430  TCGA-CM-6166-01A-11D-1650-10          7   \n",
       "311         APC  ENST00000257430  TCGA-AA-3947-01A-01D-1981-10          5   \n",
       "\n",
       "     mc3_count  diff  \n",
       "73           0     9  \n",
       "130          0     7  \n",
       "131          0     7  \n",
       "145          0     7  \n",
       "311          0     5  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_per_tx_sample[df_per_tx_sample['hugo_symbol'] == 'APC'].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.set_option('display.max_columns', None)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## TP53"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Check if there is any flag for the variants of TP53."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "sample = 'TCGA-BH-A0EE-01A-11W-A050-09'\n",
    "gene = 'TP53'\n",
    "df_gdc = pd.read_sql(\n",
    "    '''\n",
    "    SELECT \n",
    "        chromosome, start_position, end_position, \n",
    "        variant_classification, variant_type, \n",
    "        reference_allele, tumor_seq_allele1, tumor_seq_allele2,\n",
    "        dbsnp_rs, sequencer,\n",
    "        t_depth_per_caller, t_ref_count_per_caller, t_alt_count_per_caller,\n",
    "        n_depth_per_caller, callers\n",
    "    FROM gdc_grouped_callers\n",
    "    WHERE hugo_symbol=? AND tumor_sample_barcode=?\n",
    "    ''',\n",
    "    conn,\n",
    "    params=[gene, sample]\n",
    ").sort_values(['start_position', 'end_position', \"callers\"])\n",
    "df_mc3 = pd.read_sql(\n",
    "    '''\n",
    "    SELECT \n",
    "        chromosome, start_position, end_position, \n",
    "        variant_classification, variant_type, \n",
    "        reference_allele, tumor_seq_allele1, tumor_seq_allele2,\n",
    "        dbsnp_rs,\n",
    "        t_depth, t_ref_count, t_alt_count, n_depth,\n",
    "        centers, ncallers\n",
    "    FROM mc3_selected\n",
    "    WHERE hugo_symbol=? AND tumor_sample_barcode=?\n",
    "    ''',\n",
    "    conn,\n",
    "    params=[gene, sample]\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start_position</th>\n",
       "      <th>end_position</th>\n",
       "      <th>variant_classification</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>reference_allele</th>\n",
       "      <th>tumor_seq_allele1</th>\n",
       "      <th>tumor_seq_allele2</th>\n",
       "      <th>dbsnp_rs</th>\n",
       "      <th>sequencer</th>\n",
       "      <th>t_depth_per_caller</th>\n",
       "      <th>t_ref_count_per_caller</th>\n",
       "      <th>t_alt_count_per_caller</th>\n",
       "      <th>n_depth_per_caller</th>\n",
       "      <th>callers</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7675079</td>\n",
       "      <td>7675079</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td></td>\n",
       "      <td>Illumina Genome Analyzer II</td>\n",
       "      <td>86,99</td>\n",
       "      <td>58,52</td>\n",
       "      <td>25,16</td>\n",
       "      <td>167,145</td>\n",
       "      <td>somaticsniper|varscan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7675079</td>\n",
       "      <td>7675079</td>\n",
       "      <td>Frame_Shift_Del</td>\n",
       "      <td>DEL</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>novel</td>\n",
       "      <td>Illumina Genome Analyzer II</td>\n",
       "      <td>109</td>\n",
       "      <td>76</td>\n",
       "      <td>32</td>\n",
       "      <td>145</td>\n",
       "      <td>varscan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7675085</td>\n",
       "      <td>7675085</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td></td>\n",
       "      <td>Illumina Genome Analyzer II</td>\n",
       "      <td>115,115,96</td>\n",
       "      <td>66,66,50</td>\n",
       "      <td>45,45,40</td>\n",
       "      <td>153,153,131</td>\n",
       "      <td>muse|somaticsniper|varscan</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome  start_position  end_position variant_classification  \\\n",
       "1      chr17         7675079       7675079      Missense_Mutation   \n",
       "0      chr17         7675079       7675079        Frame_Shift_Del   \n",
       "2      chr17         7675085       7675085      Missense_Mutation   \n",
       "\n",
       "  variant_type reference_allele tumor_seq_allele1 tumor_seq_allele2 dbsnp_rs  \\\n",
       "1          SNP                T                 T                 G            \n",
       "0          DEL                T                 T                 -    novel   \n",
       "2          SNP                C                 C                 G            \n",
       "\n",
       "                     sequencer t_depth_per_caller t_ref_count_per_caller  \\\n",
       "1  Illumina Genome Analyzer II              86,99                  58,52   \n",
       "0  Illumina Genome Analyzer II                109                     76   \n",
       "2  Illumina Genome Analyzer II         115,115,96               66,66,50   \n",
       "\n",
       "  t_alt_count_per_caller n_depth_per_caller                     callers  \n",
       "1                  25,16            167,145       somaticsniper|varscan  \n",
       "0                     32                145                     varscan  \n",
       "2               45,45,40        153,153,131  muse|somaticsniper|varscan  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gdc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start_position</th>\n",
       "      <th>end_position</th>\n",
       "      <th>variant_classification</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>reference_allele</th>\n",
       "      <th>tumor_seq_allele1</th>\n",
       "      <th>tumor_seq_allele2</th>\n",
       "      <th>dbsnp_rs</th>\n",
       "      <th>t_depth</th>\n",
       "      <th>t_ref_count</th>\n",
       "      <th>t_alt_count</th>\n",
       "      <th>n_depth</th>\n",
       "      <th>centers</th>\n",
       "      <th>ncallers</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>17</td>\n",
       "      <td>7675079</td>\n",
       "      <td>7675085</td>\n",
       "      <td>Frame_Shift_Del</td>\n",
       "      <td>DEL</td>\n",
       "      <td>TGGGGGC</td>\n",
       "      <td>TGGGGGC</td>\n",
       "      <td>GGGGGG</td>\n",
       "      <td>novel</td>\n",
       "      <td>67</td>\n",
       "      <td>37</td>\n",
       "      <td>30</td>\n",
       "      <td>69</td>\n",
       "      <td>PINDEL|MUSE*|SOMATICSNIPER*|VARSCANI*|INDELOCA...</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome  start_position  end_position variant_classification  \\\n",
       "0         17         7675079       7675085        Frame_Shift_Del   \n",
       "\n",
       "  variant_type reference_allele tumor_seq_allele1 tumor_seq_allele2 dbsnp_rs  \\\n",
       "0          DEL          TGGGGGC           TGGGGGC            GGGGGG    novel   \n",
       "\n",
       "  t_depth t_ref_count t_alt_count n_depth  \\\n",
       "0      67          37          30      69   \n",
       "\n",
       "                                             centers ncallers  \n",
       "0  PINDEL|MUSE*|SOMATICSNIPER*|VARSCANI*|INDELOCA...        7  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_mc3"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Exact Overlap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "tx_id = 'ENST00000269305'\n",
    "df_exact_overlap = pd.read_sql(\n",
    "f'''\\\n",
    "WITH gdc_of_a_tx AS (\n",
    "    SELECT \n",
    "        chromosome, start_position, end_position, tumor_sample_barcode,\n",
    "        variant_classification, variant_type, \n",
    "        reference_allele, tumor_seq_allele1, tumor_seq_allele2,\n",
    "        t_depth_per_caller, t_ref_count_per_caller, t_alt_count_per_caller,\n",
    "        n_depth_per_caller, callers\n",
    "    FROM gdc_grp_shared_samples\n",
    "    WHERE transcript_id='{tx_id}'\n",
    "), mc3_of_a_tx AS (\n",
    "    SELECT \n",
    "        'chr' || chromosome AS chromosome, start_position, end_position, tumor_sample_barcode,\n",
    "        variant_classification, variant_type, \n",
    "        reference_allele, tumor_seq_allele1, tumor_seq_allele2,\n",
    "        t_depth, t_ref_count, t_alt_count, n_depth,\n",
    "        centers, ncallers\n",
    "    FROM mc3_selected\n",
    "    WHERE transcript_id='{tx_id}'\n",
    ") \n",
    "SELECT \n",
    "    start_position, end_position, tumor_sample_barcode, reference_allele, tumor_seq_allele2,\n",
    "    g.tumor_seq_allele1 AS gdc_tumor_seq_allele1, \n",
    "    m.tumor_seq_allele1 AS mc3_tumor_seq_allele1,\n",
    "    g.variant_classification AS gdc_variant_classification, \n",
    "    m.variant_classification AS mc3_variant_classification, \n",
    "    g.variant_type AS gdc_variant_type,\n",
    "    m.variant_type AS mc3_variant_type,\n",
    "    g.callers AS gdc_callers,\n",
    "    m.centers AS mc3_callers, \n",
    "    m.ncallers AS mc3_ncallers\n",
    "FROM gdc_of_a_tx g\n",
    "LEFT JOIN mc3_of_a_tx m\n",
    "    USING (tumor_sample_barcode, start_position, end_position, reference_allele, tumor_seq_allele2)\n",
    "UNION ALL\n",
    "SELECT \n",
    "    start_position, end_position, tumor_sample_barcode, reference_allele, tumor_seq_allele2,\n",
    "    g.tumor_seq_allele1 AS gdc_tumor_seq_allele1, \n",
    "    m.tumor_seq_allele1 AS mc3_tumor_seq_allele1,\n",
    "    g.variant_classification AS gdc_variant_classification, \n",
    "    m.variant_classification AS mc3_variant_classification, \n",
    "    g.variant_type AS gdc_variant_type,\n",
    "    m.variant_type AS mc3_variant_type,\n",
    "    g.callers AS gdc_callers,\n",
    "    m.centers AS mc3_callers, \n",
    "    m.ncallers AS mc3_ncallers\n",
    "FROM mc3_of_a_tx m\n",
    "LEFT JOIN gdc_of_a_tx g\n",
    "    USING (tumor_sample_barcode, start_position, end_position, reference_allele, tumor_seq_allele2)\n",
    "WHERE gdc_tumor_seq_allele1 IS NULL\n",
    "''', conn)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_exact_overlap = df_exact_overlap.sort_values(['start_position'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_exact_overlap = df_exact_overlap[~df_exact_overlap.duplicated()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>start_position</th>\n",
       "      <th>end_position</th>\n",
       "      <th>tumor_sample_barcode</th>\n",
       "      <th>reference_allele</th>\n",
       "      <th>tumor_seq_allele2</th>\n",
       "      <th>gdc_tumor_seq_allele1</th>\n",
       "      <th>mc3_tumor_seq_allele1</th>\n",
       "      <th>gdc_variant_classification</th>\n",
       "      <th>mc3_variant_classification</th>\n",
       "      <th>gdc_variant_type</th>\n",
       "      <th>mc3_variant_type</th>\n",
       "      <th>gdc_callers</th>\n",
       "      <th>mc3_callers</th>\n",
       "      <th>mc3_ncallers</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>354</th>\n",
       "      <td>7669353</td>\n",
       "      <td>7669353</td>\n",
       "      <td>TCGA-29-1691-01A-01W-0633-09</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>None</td>\n",
       "      <td>3'UTR</td>\n",
       "      <td>None</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>mutect</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>773</th>\n",
       "      <td>7669464</td>\n",
       "      <td>7669464</td>\n",
       "      <td>TCGA-AD-6889-01A-11D-1924-10</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>C</td>\n",
       "      <td>None</td>\n",
       "      <td>3'UTR</td>\n",
       "      <td>None</td>\n",
       "      <td>DEL</td>\n",
       "      <td>None</td>\n",
       "      <td>mutect</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1081</th>\n",
       "      <td>7669511</td>\n",
       "      <td>7669511</td>\n",
       "      <td>TCGA-D8-A1XK-01A-21D-A14K-09</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>3'UTR</td>\n",
       "      <td>3'UTR</td>\n",
       "      <td>SNP</td>\n",
       "      <td>SNP</td>\n",
       "      <td>muse</td>\n",
       "      <td>MUTECT|MUSE</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>533</th>\n",
       "      <td>7669624</td>\n",
       "      <td>7669624</td>\n",
       "      <td>TCGA-61-2113-01A-01W-0722-08</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>None</td>\n",
       "      <td>Silent</td>\n",
       "      <td>None</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>mutect</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>285</th>\n",
       "      <td>7670664</td>\n",
       "      <td>7670664</td>\n",
       "      <td>TCGA-24-1843-01A-01W-0639-09</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>Nonsense_Mutation</td>\n",
       "      <td>Nonsense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>SNP</td>\n",
       "      <td>muse|mutect|somaticsniper</td>\n",
       "      <td>SOMATICSNIPER|RADIA|MUTECT|MUSE|VARSCANS</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      start_position  end_position          tumor_sample_barcode  \\\n",
       "354          7669353       7669353  TCGA-29-1691-01A-01W-0633-09   \n",
       "773          7669464       7669464  TCGA-AD-6889-01A-11D-1924-10   \n",
       "1081         7669511       7669511  TCGA-D8-A1XK-01A-21D-A14K-09   \n",
       "533          7669624       7669624  TCGA-61-2113-01A-01W-0722-08   \n",
       "285          7670664       7670664  TCGA-24-1843-01A-01W-0639-09   \n",
       "\n",
       "     reference_allele tumor_seq_allele2 gdc_tumor_seq_allele1  \\\n",
       "354                 T                 C                     T   \n",
       "773                 C                 -                     C   \n",
       "1081                T                 C                     T   \n",
       "533                 C                 T                     C   \n",
       "285                 C                 A                     C   \n",
       "\n",
       "     mc3_tumor_seq_allele1 gdc_variant_classification  \\\n",
       "354                   None                      3'UTR   \n",
       "773                   None                      3'UTR   \n",
       "1081                     T                      3'UTR   \n",
       "533                   None                     Silent   \n",
       "285                      C          Nonsense_Mutation   \n",
       "\n",
       "     mc3_variant_classification gdc_variant_type mc3_variant_type  \\\n",
       "354                        None              SNP             None   \n",
       "773                        None              DEL             None   \n",
       "1081                      3'UTR              SNP              SNP   \n",
       "533                        None              SNP             None   \n",
       "285           Nonsense_Mutation              SNP              SNP   \n",
       "\n",
       "                    gdc_callers                               mc3_callers  \\\n",
       "354                      mutect                                      None   \n",
       "773                      mutect                                      None   \n",
       "1081                       muse                               MUTECT|MUSE   \n",
       "533                      mutect                                      None   \n",
       "285   muse|mutect|somaticsniper  SOMATICSNIPER|RADIA|MUTECT|MUSE|VARSCANS   \n",
       "\n",
       "     mc3_ncallers  \n",
       "354          None  \n",
       "773          None  \n",
       "1081            2  \n",
       "533          None  \n",
       "285             5  "
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_exact_overlap.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "def grp_fun(df):\n",
    "    gdc_called = df['gdc_callers'].notnull()\n",
    "    mc3_called = df['mc3_callers'].notnull()\n",
    "    return pd.DataFrame([[ \n",
    "        (gdc_called & mc3_called).sum(),  \n",
    "        gdc_called.sum() - (gdc_called & mc3_called).sum(),\n",
    "        mc3_called.sum() - (gdc_called & mc3_called).sum()\n",
    "    ]], columns=['shared', 'only_in_gdc', 'only_in_mc3'])\n",
    "df_overlap_grp = df_exact_overlap.groupby(['start_position', 'end_position']).apply(grp_fun).reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>start_position</th>\n",
       "      <th>end_position</th>\n",
       "      <th>level_2</th>\n",
       "      <th>shared</th>\n",
       "      <th>only_in_gdc</th>\n",
       "      <th>only_in_mc3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>7669353</td>\n",
       "      <td>7669353</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>7669464</td>\n",
       "      <td>7669464</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>7669624</td>\n",
       "      <td>7669624</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>7673243</td>\n",
       "      <td>7673243</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>7673579</td>\n",
       "      <td>7673579</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>7673755</td>\n",
       "      <td>7673756</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>7673756</td>\n",
       "      <td>7673757</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>7673763</td>\n",
       "      <td>7673763</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>7673775</td>\n",
       "      <td>7673776</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>7673783</td>\n",
       "      <td>7673796</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>7673788</td>\n",
       "      <td>7673788</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>7673801</td>\n",
       "      <td>7673810</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>7673803</td>\n",
       "      <td>7673803</td>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>7673835</td>\n",
       "      <td>7673835</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>7674174</td>\n",
       "      <td>7674183</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>7674182</td>\n",
       "      <td>7674183</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>7674185</td>\n",
       "      <td>7674185</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>123</th>\n",
       "      <td>7674206</td>\n",
       "      <td>7674206</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161</th>\n",
       "      <td>7674276</td>\n",
       "      <td>7674277</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>166</th>\n",
       "      <td>7674694</td>\n",
       "      <td>7674695</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>167</th>\n",
       "      <td>7674779</td>\n",
       "      <td>7674780</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>172</th>\n",
       "      <td>7674863</td>\n",
       "      <td>7674864</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180</th>\n",
       "      <td>7674886</td>\n",
       "      <td>7674886</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>195</th>\n",
       "      <td>7674927</td>\n",
       "      <td>7675013</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>205</th>\n",
       "      <td>7674948</td>\n",
       "      <td>7674972</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>209</th>\n",
       "      <td>7674957</td>\n",
       "      <td>7674957</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>211</th>\n",
       "      <td>7674960</td>\n",
       "      <td>7674961</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>221</th>\n",
       "      <td>7675075</td>\n",
       "      <td>7675075</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>222</th>\n",
       "      <td>7675076</td>\n",
       "      <td>7675076</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>225</th>\n",
       "      <td>7675079</td>\n",
       "      <td>7675079</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>230</th>\n",
       "      <td>7675085</td>\n",
       "      <td>7675085</td>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>237</th>\n",
       "      <td>7675095</td>\n",
       "      <td>7675095</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>238</th>\n",
       "      <td>7675096</td>\n",
       "      <td>7675113</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>242</th>\n",
       "      <td>7675111</td>\n",
       "      <td>7675111</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>260</th>\n",
       "      <td>7675142</td>\n",
       "      <td>7675143</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>267</th>\n",
       "      <td>7675147</td>\n",
       "      <td>7675147</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>272</th>\n",
       "      <td>7675152</td>\n",
       "      <td>7675182</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>285</th>\n",
       "      <td>7675181</td>\n",
       "      <td>7675182</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>286</th>\n",
       "      <td>7675182</td>\n",
       "      <td>7675182</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>287</th>\n",
       "      <td>7675184</td>\n",
       "      <td>7675185</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>311</th>\n",
       "      <td>7675229</td>\n",
       "      <td>7675229</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>313</th>\n",
       "      <td>7675231</td>\n",
       "      <td>7675241</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>314</th>\n",
       "      <td>7675232</td>\n",
       "      <td>7675232</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320</th>\n",
       "      <td>7675239</td>\n",
       "      <td>7675239</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>321</th>\n",
       "      <td>7675246</td>\n",
       "      <td>7675246</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>323</th>\n",
       "      <td>7675994</td>\n",
       "      <td>7675994</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>325</th>\n",
       "      <td>7675997</td>\n",
       "      <td>7675997</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>330</th>\n",
       "      <td>7676025</td>\n",
       "      <td>7676033</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>342</th>\n",
       "      <td>7676046</td>\n",
       "      <td>7676046</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345</th>\n",
       "      <td>7676049</td>\n",
       "      <td>7676049</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>346</th>\n",
       "      <td>7676049</td>\n",
       "      <td>7676050</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>350</th>\n",
       "      <td>7676065</td>\n",
       "      <td>7676065</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>356</th>\n",
       "      <td>7676097</td>\n",
       "      <td>7676097</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>366</th>\n",
       "      <td>7676153</td>\n",
       "      <td>7676153</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>367</th>\n",
       "      <td>7676171</td>\n",
       "      <td>7676172</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>372</th>\n",
       "      <td>7676208</td>\n",
       "      <td>7676218</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>379</th>\n",
       "      <td>7676251</td>\n",
       "      <td>7676252</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>383</th>\n",
       "      <td>7676267</td>\n",
       "      <td>7676280</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>389</th>\n",
       "      <td>7676500</td>\n",
       "      <td>7676500</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>391</th>\n",
       "      <td>7676604</td>\n",
       "      <td>7676604</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>61 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     start_position  end_position  level_2  shared  only_in_gdc  only_in_mc3\n",
       "0           7669353       7669353        0       0            1            0\n",
       "1           7669464       7669464        0       0            1            0\n",
       "3           7669624       7669624        0       0            1            0\n",
       "23          7673243       7673243        0       0            1            0\n",
       "36          7673579       7673579        0       2            1            0\n",
       "56          7673755       7673756        0       0            1            0\n",
       "57          7673756       7673757        0       0            1            0\n",
       "60          7673763       7673763        0       4            1            0\n",
       "66          7673775       7673776        0       0            1            0\n",
       "75          7673783       7673796        0       0            1            0\n",
       "78          7673788       7673788        0       8            1            0\n",
       "86          7673801       7673810        0       0            1            0\n",
       "88          7673803       7673803        0      21            1            1\n",
       "105         7673835       7673835        0       1            1            0\n",
       "107         7674174       7674183        0       0            1            0\n",
       "109         7674182       7674183        0       0            1            0\n",
       "111         7674185       7674185        0       0            1            0\n",
       "123         7674206       7674206        0       2            1            0\n",
       "161         7674276       7674277        0       1            1            0\n",
       "166         7674694       7674695        0       0            1            0\n",
       "167         7674779       7674780        0       0            1            0\n",
       "172         7674863       7674864        0       0            1            0\n",
       "180         7674886       7674886        0       1            1            0\n",
       "195         7674927       7675013        0       0            1            0\n",
       "205         7674948       7674972        0       0            1            0\n",
       "209         7674957       7674957        0      11            2            0\n",
       "211         7674960       7674961        0       0            1            0\n",
       "221         7675075       7675075        0       0            1            0\n",
       "222         7675076       7675076        0      13            1            0\n",
       "225         7675079       7675079        0       0            2            0\n",
       "..              ...           ...      ...     ...          ...          ...\n",
       "230         7675085       7675085        0      21            2            0\n",
       "237         7675095       7675095        0       9            1            0\n",
       "238         7675096       7675113        0       0            1            0\n",
       "242         7675111       7675111        0       0            2            0\n",
       "260         7675142       7675143        0       0            1            0\n",
       "267         7675147       7675147        0       0            1            0\n",
       "272         7675152       7675182        0       0            1            0\n",
       "285         7675181       7675182        0       0            1            0\n",
       "286         7675182       7675182        0       1            1            1\n",
       "287         7675184       7675185        0       0            1            0\n",
       "311         7675229       7675229        0       2            2            0\n",
       "313         7675231       7675241        0       0            1            0\n",
       "314         7675232       7675232        0       6            2            0\n",
       "320         7675239       7675239        0       0            1            0\n",
       "321         7675246       7675246        0       0            1            0\n",
       "323         7675994       7675994        0       6            2            0\n",
       "325         7675997       7675997        0       2            1            0\n",
       "330         7676025       7676033        0       0            1            0\n",
       "342         7676046       7676046        0       2            1            0\n",
       "345         7676049       7676049        0       0            1            0\n",
       "346         7676049       7676050        0       0            1            0\n",
       "350         7676065       7676065        0       1            1            0\n",
       "356         7676097       7676097        0       1            1            0\n",
       "366         7676153       7676153        0       0            1            0\n",
       "367         7676171       7676172        0       0            1            0\n",
       "372         7676208       7676218        0       0            1            0\n",
       "379         7676251       7676252        0       0            1            0\n",
       "383         7676267       7676280        0       0            1            0\n",
       "389         7676500       7676500        0       0            1            0\n",
       "391         7676604       7676604        0       0            1            0\n",
       "\n",
       "[61 rows x 6 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_overlap_grp[df_overlap_grp['only_in_gdc'] > 0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>start_position</th>\n",
       "      <th>end_position</th>\n",
       "      <th>level_2</th>\n",
       "      <th>shared</th>\n",
       "      <th>only_in_gdc</th>\n",
       "      <th>only_in_mc3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>7673776</td>\n",
       "      <td>7673778</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>7673803</td>\n",
       "      <td>7673803</td>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>7673806</td>\n",
       "      <td>7673806</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>7673813</td>\n",
       "      <td>7673814</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>7674185</td>\n",
       "      <td>7674186</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210</th>\n",
       "      <td>7674957</td>\n",
       "      <td>7674959</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>226</th>\n",
       "      <td>7675079</td>\n",
       "      <td>7675085</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>262</th>\n",
       "      <td>7675143</td>\n",
       "      <td>7675147</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270</th>\n",
       "      <td>7675149</td>\n",
       "      <td>7675182</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>286</th>\n",
       "      <td>7675182</td>\n",
       "      <td>7675182</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>300</th>\n",
       "      <td>7675208</td>\n",
       "      <td>7675208</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>302</th>\n",
       "      <td>7675212</td>\n",
       "      <td>7675212</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>312</th>\n",
       "      <td>7675231</td>\n",
       "      <td>7675232</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>341</th>\n",
       "      <td>7676044</td>\n",
       "      <td>7676078</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>344</th>\n",
       "      <td>7676047</td>\n",
       "      <td>7676049</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>368</th>\n",
       "      <td>7676172</td>\n",
       "      <td>7676173</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     start_position  end_position  level_2  shared  only_in_gdc  only_in_mc3\n",
       "68          7673776       7673778        0       0            0            1\n",
       "88          7673803       7673803        0      21            1            1\n",
       "90          7673806       7673806        0      12            0            1\n",
       "95          7673813       7673814        0       0            0            1\n",
       "112         7674185       7674186        0       0            0            1\n",
       "210         7674957       7674959        0       2            0            1\n",
       "226         7675079       7675085        0       0            0            1\n",
       "262         7675143       7675147        0       0            0            1\n",
       "270         7675149       7675182        0       0            0            1\n",
       "286         7675182       7675182        0       1            1            1\n",
       "300         7675208       7675208        0       4            0            2\n",
       "302         7675212       7675212        0       0            0            1\n",
       "312         7675231       7675232        0       0            0            1\n",
       "341         7676044       7676078        0       0            0            1\n",
       "344         7676047       7676049        0       0            0            1\n",
       "368         7676172       7676173        0       0            0            1"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_overlap_grp[df_overlap_grp['only_in_mc3'] > 0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_overlap_grp.to_csv('../processed_data/tp53_overlap.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
