{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sqlite3\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "conn = sqlite3.connect('../variants.sqlite')\n",
    "conn.executescript('''\\\n",
    "PRAGMA cache_size=-4192000;\n",
    "PRAGMA temp_store=MEMORY;\n",
    "PRAGMA journal_mode=OFF;\n",
    "''')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_variant_class = pd.merge(\n",
    "    pd.read_sql('''\\\n",
    "SELECT variant_classification, count(*) AS 'mc3_count'\n",
    "FROM mc3_selected\n",
    "GROUP BY variant_classification\n",
    "ORDER BY mc3_count DESC\n",
    "''', conn),\n",
    "    pd.read_sql('''\\\n",
    "SELECT variant_classification, count(*) AS 'gdc_count'\n",
    "FROM gdc_grouped_callers\n",
    "GROUP BY variant_classification\n",
    "ORDER BY gdc_count DESC\n",
    "''', conn),\n",
    "    how='outer',\n",
    "    on='variant_classification'\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "variant_classification    Missense_MutationSilent3'UTRFrame_Shift_DelNon...\n",
       "mc3_count                                                            445856\n",
       "gdc_count                                                            515859\n",
       "dtype: object"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_variant_class.sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_per_tx_sample_variant_type = pd.read_sql('''\\\n",
    "WITH gdc_by_tx AS (\n",
    "    SELECT hugo_symbol, transcript_id, variant_classification, tumor_sample_barcode, count(*) AS 'gdc_count'\n",
    "    FROM gdc_grouped_callers\n",
    "    GROUP BY hugo_symbol, transcript_id, variant_classification, tumor_sample_barcode\n",
    "), mc3_by_tx AS (\n",
    "    SELECT hugo_symbol, transcript_id, variant_classification, tumor_sample_barcode, count(*) AS 'mc3_count'\n",
    "    FROM mc3_selected\n",
    "    GROUP BY hugo_symbol, transcript_id, variant_classification, tumor_sample_barcode\n",
    ")\n",
    "SELECT hugo_symbol, transcript_id, variant_classification, tumor_sample_barcode, gdc_count, mc3_count FROM gdc_by_tx\n",
    "LEFT JOIN mc3_by_tx \n",
    "USING (hugo_symbol, transcript_id, variant_classification, tumor_sample_barcode)\n",
    "UNION ALL\n",
    "SELECT hugo_symbol, transcript_id, variant_classification, tumor_sample_barcode, gdc_count, mc3_count FROM mc3_by_tx\n",
    "LEFT JOIN gdc_by_tx\n",
    "USING (hugo_symbol, transcript_id, variant_classification, tumor_sample_barcode)\n",
    "WHERE gdc_by_tx.gdc_count IS NULL\n",
    "''', conn)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_per_tx_sample_variant_type = (\n",
    "    df_per_tx_sample_variant_type\n",
    "    .fillna(0)\n",
    "    .assign(\n",
    "        gdc_count=lambda df: df.gdc_count.astype(int),\n",
    "        mc3_count=lambda df: df.mc3_count.astype(int))\n",
    "    .assign(\n",
    "        diff=lambda df: df.gdc_count-df.mc3_count\n",
    "    )\n",
    "    .sort_values(['diff'], ascending=False)\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hugo_symbol</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>variant_classification</th>\n",
       "      <th>tumor_sample_barcode</th>\n",
       "      <th>gdc_count</th>\n",
       "      <th>mc3_count</th>\n",
       "      <th>diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>438110</th>\n",
       "      <td>TTN</td>\n",
       "      <td>ENST00000591111</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>TCGA-CA-6717-01A-11D-1835-10</td>\n",
       "      <td>29</td>\n",
       "      <td>0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>437991</th>\n",
       "      <td>TTN</td>\n",
       "      <td>ENST00000591111</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>TCGA-AA-A010-01A-01D-A17O-10</td>\n",
       "      <td>29</td>\n",
       "      <td>0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>438021</th>\n",
       "      <td>TTN</td>\n",
       "      <td>ENST00000591111</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>TCGA-AN-A046-01A-21W-A050-09</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>437990</th>\n",
       "      <td>TTN</td>\n",
       "      <td>ENST00000591111</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>TCGA-AA-A00N-01A-02D-A17O-10</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>438059</th>\n",
       "      <td>TTN</td>\n",
       "      <td>ENST00000591111</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>TCGA-AZ-4315-01A-01D-1408-10</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       hugo_symbol    transcript_id variant_classification  \\\n",
       "438110         TTN  ENST00000591111      Missense_Mutation   \n",
       "437991         TTN  ENST00000591111      Missense_Mutation   \n",
       "438021         TTN  ENST00000591111      Missense_Mutation   \n",
       "437990         TTN  ENST00000591111      Missense_Mutation   \n",
       "438059         TTN  ENST00000591111      Missense_Mutation   \n",
       "\n",
       "                tumor_sample_barcode  gdc_count  mc3_count  diff  \n",
       "438110  TCGA-CA-6717-01A-11D-1835-10         29          0    29  \n",
       "437991  TCGA-AA-A010-01A-01D-A17O-10         29          0    29  \n",
       "438021  TCGA-AN-A046-01A-21W-A050-09         28          0    28  \n",
       "437990  TCGA-AA-A00N-01A-02D-A17O-10         28          0    28  \n",
       "438059  TCGA-AZ-4315-01A-01D-1408-10         23          0    23  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_per_tx_sample_variant_type.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hugo_symbol</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>variant_classification</th>\n",
       "      <th>tumor_sample_barcode</th>\n",
       "      <th>gdc_count</th>\n",
       "      <th>mc3_count</th>\n",
       "      <th>diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>427205</th>\n",
       "      <td>TP53</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Splice_Region</td>\n",
       "      <td>TCGA-13-0755-01A-01W-0371-08</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>592756</th>\n",
       "      <td>TP53</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Silent</td>\n",
       "      <td>TCGA-13-0755-01A-01W-0371-08</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       hugo_symbol    transcript_id variant_classification  \\\n",
       "427205        TP53  ENST00000269305          Splice_Region   \n",
       "592756        TP53  ENST00000269305                 Silent   \n",
       "\n",
       "                tumor_sample_barcode  gdc_count  mc3_count  diff  \n",
       "427205  TCGA-13-0755-01A-01W-0371-08          2          0     2  \n",
       "592756  TCGA-13-0755-01A-01W-0371-08          0          1    -1  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_per_tx_sample_variant_type[\n",
    "    (df_per_tx_sample_variant_type['hugo_symbol'] == 'TP53') &\n",
    "    (df_per_tx_sample_variant_type['tumor_sample_barcode'] == 'TCGA-13-0755-01A-01W-0371-08')\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_per_tx_sample = (\n",
    "    df_per_tx_sample_variant_type\n",
    "    .groupby(['hugo_symbol', 'transcript_id', 'tumor_sample_barcode'])\n",
    "    .sum()\n",
    "    .sort_values(['diff'], ascending=False)\n",
    "    .reset_index()\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hugo_symbol</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>tumor_sample_barcode</th>\n",
       "      <th>gdc_count</th>\n",
       "      <th>mc3_count</th>\n",
       "      <th>diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TTN</td>\n",
       "      <td>ENST00000591111</td>\n",
       "      <td>TCGA-CA-6717-01A-11D-1835-10</td>\n",
       "      <td>68</td>\n",
       "      <td>0</td>\n",
       "      <td>68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TTN</td>\n",
       "      <td>ENST00000591111</td>\n",
       "      <td>TCGA-AA-A010-01A-01D-A17O-10</td>\n",
       "      <td>48</td>\n",
       "      <td>0</td>\n",
       "      <td>48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TTN</td>\n",
       "      <td>ENST00000591111</td>\n",
       "      <td>TCGA-AN-A046-01A-21W-A050-09</td>\n",
       "      <td>45</td>\n",
       "      <td>0</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TTN</td>\n",
       "      <td>ENST00000591111</td>\n",
       "      <td>TCGA-AZ-4315-01A-01D-1408-10</td>\n",
       "      <td>44</td>\n",
       "      <td>0</td>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TTN</td>\n",
       "      <td>ENST00000591111</td>\n",
       "      <td>TCGA-AA-A00N-01A-02D-A17O-10</td>\n",
       "      <td>40</td>\n",
       "      <td>0</td>\n",
       "      <td>40</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  hugo_symbol    transcript_id          tumor_sample_barcode  gdc_count  \\\n",
       "0         TTN  ENST00000591111  TCGA-CA-6717-01A-11D-1835-10         68   \n",
       "1         TTN  ENST00000591111  TCGA-AA-A010-01A-01D-A17O-10         48   \n",
       "2         TTN  ENST00000591111  TCGA-AN-A046-01A-21W-A050-09         45   \n",
       "3         TTN  ENST00000591111  TCGA-AZ-4315-01A-01D-1408-10         44   \n",
       "4         TTN  ENST00000591111  TCGA-AA-A00N-01A-02D-A17O-10         40   \n",
       "\n",
       "   mc3_count  diff  \n",
       "0          0    68  \n",
       "1          0    48  \n",
       "2          0    45  \n",
       "3          0    44  \n",
       "4          0    40  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_per_tx_sample.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_per_tx = (\n",
    "    df_per_tx_sample.groupby(['hugo_symbol', 'transcript_id'])\n",
    "    .sum()\n",
    "    .sort_values(['diff'], ascending=False)\n",
    "    .reset_index()\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hugo_symbol</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gdc_count</th>\n",
       "      <th>mc3_count</th>\n",
       "      <th>diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TTN</td>\n",
       "      <td>ENST00000591111</td>\n",
       "      <td>1901</td>\n",
       "      <td>0</td>\n",
       "      <td>1901</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>APC</td>\n",
       "      <td>ENST00000257430</td>\n",
       "      <td>616</td>\n",
       "      <td>0</td>\n",
       "      <td>616</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DST</td>\n",
       "      <td>ENST00000312431</td>\n",
       "      <td>379</td>\n",
       "      <td>0</td>\n",
       "      <td>379</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OBSCN</td>\n",
       "      <td>ENST00000422127</td>\n",
       "      <td>378</td>\n",
       "      <td>0</td>\n",
       "      <td>378</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MUC4</td>\n",
       "      <td>ENST00000463781</td>\n",
       "      <td>479</td>\n",
       "      <td>145</td>\n",
       "      <td>334</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NEB</td>\n",
       "      <td>ENST00000172853</td>\n",
       "      <td>309</td>\n",
       "      <td>0</td>\n",
       "      <td>309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>TP53</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>1291</td>\n",
       "      <td>992</td>\n",
       "      <td>299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>FAT3</td>\n",
       "      <td>ENST00000525166</td>\n",
       "      <td>291</td>\n",
       "      <td>0</td>\n",
       "      <td>291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MACF1</td>\n",
       "      <td>ENST00000372915</td>\n",
       "      <td>274</td>\n",
       "      <td>0</td>\n",
       "      <td>274</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>CSMD1</td>\n",
       "      <td>ENST00000520002</td>\n",
       "      <td>265</td>\n",
       "      <td>0</td>\n",
       "      <td>265</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>SYNE2</td>\n",
       "      <td>ENST00000344113</td>\n",
       "      <td>260</td>\n",
       "      <td>0</td>\n",
       "      <td>260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>DNAH11</td>\n",
       "      <td>ENST00000409508</td>\n",
       "      <td>256</td>\n",
       "      <td>0</td>\n",
       "      <td>256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ADGRV1</td>\n",
       "      <td>ENST00000405460</td>\n",
       "      <td>233</td>\n",
       "      <td>0</td>\n",
       "      <td>233</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>XIRP2</td>\n",
       "      <td>ENST00000628543</td>\n",
       "      <td>215</td>\n",
       "      <td>0</td>\n",
       "      <td>215</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>CCDC168</td>\n",
       "      <td>ENST00000322527</td>\n",
       "      <td>303</td>\n",
       "      <td>93</td>\n",
       "      <td>210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>SACS</td>\n",
       "      <td>ENST00000382292</td>\n",
       "      <td>206</td>\n",
       "      <td>0</td>\n",
       "      <td>206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>FCGBP</td>\n",
       "      <td>ENST00000616721</td>\n",
       "      <td>196</td>\n",
       "      <td>0</td>\n",
       "      <td>196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>MUC2</td>\n",
       "      <td>ENST00000361558</td>\n",
       "      <td>190</td>\n",
       "      <td>0</td>\n",
       "      <td>190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>GATA3</td>\n",
       "      <td>ENST00000346208</td>\n",
       "      <td>186</td>\n",
       "      <td>0</td>\n",
       "      <td>186</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>DCHS2</td>\n",
       "      <td>ENST00000623607</td>\n",
       "      <td>172</td>\n",
       "      <td>0</td>\n",
       "      <td>172</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   hugo_symbol    transcript_id  gdc_count  mc3_count  diff\n",
       "0          TTN  ENST00000591111       1901          0  1901\n",
       "1          APC  ENST00000257430        616          0   616\n",
       "2          DST  ENST00000312431        379          0   379\n",
       "3        OBSCN  ENST00000422127        378          0   378\n",
       "4         MUC4  ENST00000463781        479        145   334\n",
       "5          NEB  ENST00000172853        309          0   309\n",
       "6         TP53  ENST00000269305       1291        992   299\n",
       "7         FAT3  ENST00000525166        291          0   291\n",
       "8        MACF1  ENST00000372915        274          0   274\n",
       "9        CSMD1  ENST00000520002        265          0   265\n",
       "10       SYNE2  ENST00000344113        260          0   260\n",
       "11      DNAH11  ENST00000409508        256          0   256\n",
       "12      ADGRV1  ENST00000405460        233          0   233\n",
       "13       XIRP2  ENST00000628543        215          0   215\n",
       "14     CCDC168  ENST00000322527        303         93   210\n",
       "15        SACS  ENST00000382292        206          0   206\n",
       "16       FCGBP  ENST00000616721        196          0   196\n",
       "17        MUC2  ENST00000361558        190          0   190\n",
       "18       GATA3  ENST00000346208        186          0   186\n",
       "19       DCHS2  ENST00000623607        172          0   172"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_per_tx.head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Export data to CSV\n",
    "df_per_tx.to_csv('../variant_count_diff.per_tx.csv', index=False)\n",
    "df_per_tx_sample.to_csv('../variant_count_diff.per_tx_sample.csv', index=False)\n",
    "df_per_tx_sample_variant_type.to_csv('../variant_count_diff.per_tx_sample_type.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hugo_symbol</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gdc_count</th>\n",
       "      <th>mc3_count</th>\n",
       "      <th>diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3130</th>\n",
       "      <td>U2AF1L4</td>\n",
       "      <td>ENST00000412391</td>\n",
       "      <td>17</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12801</th>\n",
       "      <td>U2AF2</td>\n",
       "      <td>ENST00000308924</td>\n",
       "      <td>16</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22392</th>\n",
       "      <td>U2AF1</td>\n",
       "      <td>ENST00000291552</td>\n",
       "      <td>5</td>\n",
       "      <td>13</td>\n",
       "      <td>-8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22597</th>\n",
       "      <td>U2AF1L4</td>\n",
       "      <td>ENST00000292879</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>-9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      hugo_symbol    transcript_id  gdc_count  mc3_count  diff\n",
       "3130      U2AF1L4  ENST00000412391         17          0    17\n",
       "12801       U2AF2  ENST00000308924         16         14     2\n",
       "22392       U2AF1  ENST00000291552          5         13    -8\n",
       "22597     U2AF1L4  ENST00000292879          0          9    -9"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_per_tx[df_per_tx['hugo_symbol'].str.startswith('U2AF')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hugo_symbol</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>tumor_sample_barcode</th>\n",
       "      <th>gdc_count</th>\n",
       "      <th>mc3_count</th>\n",
       "      <th>diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>182774</th>\n",
       "      <td>U2AF1</td>\n",
       "      <td>ENST00000291552</td>\n",
       "      <td>TCGA-AN-A0AK-01A-21W-A019-09</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>182775</th>\n",
       "      <td>U2AF1</td>\n",
       "      <td>ENST00000291552</td>\n",
       "      <td>TCGA-AA-3966-01A-01D-1981-10</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>182776</th>\n",
       "      <td>U2AF1</td>\n",
       "      <td>ENST00000291552</td>\n",
       "      <td>TCGA-AA-3672-01A-01W-0900-09</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>182777</th>\n",
       "      <td>U2AF1</td>\n",
       "      <td>ENST00000291552</td>\n",
       "      <td>TCGA-A6-3809-01A-01D-A270-10</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>232319</th>\n",
       "      <td>U2AF1</td>\n",
       "      <td>ENST00000291552</td>\n",
       "      <td>TCGA-CM-5861-01A-01D-1650-10</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>475792</th>\n",
       "      <td>U2AF1</td>\n",
       "      <td>ENST00000291552</td>\n",
       "      <td>TCGA-GM-A5PX-01A-12D-A28B-09</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>475793</th>\n",
       "      <td>U2AF1</td>\n",
       "      <td>ENST00000291552</td>\n",
       "      <td>TCGA-E2-A1IO-01A-11D-A142-09</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>475794</th>\n",
       "      <td>U2AF1</td>\n",
       "      <td>ENST00000291552</td>\n",
       "      <td>TCGA-DM-A1HB-01A-21D-A183-10</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>475796</th>\n",
       "      <td>U2AF1</td>\n",
       "      <td>ENST00000291552</td>\n",
       "      <td>TCGA-CM-4744-01A-01D-1408-10</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>475797</th>\n",
       "      <td>U2AF1</td>\n",
       "      <td>ENST00000291552</td>\n",
       "      <td>TCGA-AD-6889-01A-11D-1924-10</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>475798</th>\n",
       "      <td>U2AF1</td>\n",
       "      <td>ENST00000291552</td>\n",
       "      <td>TCGA-AA-A017-01A-01W-A00E-09</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>475799</th>\n",
       "      <td>U2AF1</td>\n",
       "      <td>ENST00000291552</td>\n",
       "      <td>TCGA-AA-3715-01A-01W-0900-09</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>475800</th>\n",
       "      <td>U2AF1</td>\n",
       "      <td>ENST00000291552</td>\n",
       "      <td>TCGA-09-2044-01B-01D-0704-01</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>553311</th>\n",
       "      <td>U2AF1</td>\n",
       "      <td>ENST00000291552</td>\n",
       "      <td>TCGA-CK-4951-01A-01D-1408-10</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>553376</th>\n",
       "      <td>U2AF1</td>\n",
       "      <td>ENST00000291552</td>\n",
       "      <td>TCGA-A7-A13F-01A-11D-A12Q-09</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>-2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       hugo_symbol    transcript_id          tumor_sample_barcode  gdc_count  \\\n",
       "182774       U2AF1  ENST00000291552  TCGA-AN-A0AK-01A-21W-A019-09          1   \n",
       "182775       U2AF1  ENST00000291552  TCGA-AA-3966-01A-01D-1981-10          1   \n",
       "182776       U2AF1  ENST00000291552  TCGA-AA-3672-01A-01W-0900-09          1   \n",
       "182777       U2AF1  ENST00000291552  TCGA-A6-3809-01A-01D-A270-10          1   \n",
       "232319       U2AF1  ENST00000291552  TCGA-CM-5861-01A-01D-1650-10          1   \n",
       "475792       U2AF1  ENST00000291552  TCGA-GM-A5PX-01A-12D-A28B-09          0   \n",
       "475793       U2AF1  ENST00000291552  TCGA-E2-A1IO-01A-11D-A142-09          0   \n",
       "475794       U2AF1  ENST00000291552  TCGA-DM-A1HB-01A-21D-A183-10          0   \n",
       "475796       U2AF1  ENST00000291552  TCGA-CM-4744-01A-01D-1408-10          0   \n",
       "475797       U2AF1  ENST00000291552  TCGA-AD-6889-01A-11D-1924-10          0   \n",
       "475798       U2AF1  ENST00000291552  TCGA-AA-A017-01A-01W-A00E-09          0   \n",
       "475799       U2AF1  ENST00000291552  TCGA-AA-3715-01A-01W-0900-09          0   \n",
       "475800       U2AF1  ENST00000291552  TCGA-09-2044-01B-01D-0704-01          0   \n",
       "553311       U2AF1  ENST00000291552  TCGA-CK-4951-01A-01D-1408-10          0   \n",
       "553376       U2AF1  ENST00000291552  TCGA-A7-A13F-01A-11D-A12Q-09          0   \n",
       "\n",
       "        mc3_count  diff  \n",
       "182774          0     1  \n",
       "182775          0     1  \n",
       "182776          0     1  \n",
       "182777          0     1  \n",
       "232319          1     0  \n",
       "475792          1    -1  \n",
       "475793          1    -1  \n",
       "475794          1    -1  \n",
       "475796          1    -1  \n",
       "475797          1    -1  \n",
       "475798          1    -1  \n",
       "475799          1    -1  \n",
       "475800          1    -1  \n",
       "553311          2    -2  \n",
       "553376          2    -2  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_per_tx_sample[df_per_tx_sample['hugo_symbol'] == 'U2AF1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.set_option('display.max_columns', None)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## TP53"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Check if there is any flag for the variants of TP53."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\"group_concat(caller, '|')\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "sample = 'TCGA-BH-A0EE-01A-11W-A050-09'\n",
    "gene = 'TP53'\n",
    "df_gdc = pd.read_sql(\n",
    "    '''\n",
    "    SELECT \n",
    "        chromosome, start_position, end_position, \n",
    "        variant_classification, variant_type, \n",
    "        reference_allele, tumor_seq_allele1, tumor_seq_allele2,\n",
    "        dbsnp_rs, sequencer,\n",
    "        \"group_concat(t_depth, ',')\", \"group_concat(t_ref_count, ',')\", \"group_concat(t_alt_count, ',')\",\n",
    "        \"group_concat(n_depth, ',')\", \"group_concat(caller, '|')\"\n",
    "    FROM gdc_grouped_callers\n",
    "    WHERE hugo_symbol=? AND tumor_sample_barcode=?\n",
    "    ''',\n",
    "    conn,\n",
    "    params=[gene, sample]\n",
    ").sort_values(['start_position', 'end_position', \"group_concat(caller, '|')\"])\n",
    "df_mc3 = pd.read_sql(\n",
    "    '''\n",
    "    SELECT \n",
    "        chromosome, start_position, end_position, \n",
    "        variant_classification, variant_type, \n",
    "        reference_allele, tumor_seq_allele1, tumor_seq_allele2,\n",
    "        dbsnp_rs,\n",
    "        t_depth, t_ref_count, t_alt_count, n_depth,\n",
    "        centers, ncallers\n",
    "    FROM mc3_selected\n",
    "    WHERE hugo_symbol=? AND tumor_sample_barcode=?\n",
    "    ''',\n",
    "    conn,\n",
    "    params=[gene, sample]\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start_position</th>\n",
       "      <th>end_position</th>\n",
       "      <th>variant_classification</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>reference_allele</th>\n",
       "      <th>tumor_seq_allele1</th>\n",
       "      <th>tumor_seq_allele2</th>\n",
       "      <th>dbsnp_rs</th>\n",
       "      <th>sequencer</th>\n",
       "      <th>group_concat(t_depth, ',')</th>\n",
       "      <th>group_concat(t_ref_count, ',')</th>\n",
       "      <th>group_concat(t_alt_count, ',')</th>\n",
       "      <th>group_concat(n_depth, ',')</th>\n",
       "      <th>group_concat(caller, '|')</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7675079</td>\n",
       "      <td>7675079</td>\n",
       "      <td>Frame_Shift_Del</td>\n",
       "      <td>DEL</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>novel</td>\n",
       "      <td>Illumina Genome Analyzer II</td>\n",
       "      <td>109</td>\n",
       "      <td>76</td>\n",
       "      <td>32</td>\n",
       "      <td>145</td>\n",
       "      <td>varscan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7675079</td>\n",
       "      <td>7675079</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td></td>\n",
       "      <td>Illumina Genome Analyzer II</td>\n",
       "      <td>99,86</td>\n",
       "      <td>52,58</td>\n",
       "      <td>16,25</td>\n",
       "      <td>145,167</td>\n",
       "      <td>varscan|somaticsniper</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7675085</td>\n",
       "      <td>7675085</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td></td>\n",
       "      <td>Illumina Genome Analyzer II</td>\n",
       "      <td>96,115,115</td>\n",
       "      <td>50,66,66</td>\n",
       "      <td>40,45,45</td>\n",
       "      <td>131,153,153</td>\n",
       "      <td>varscan|somaticsniper|muse</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome start_position end_position variant_classification variant_type  \\\n",
       "0      chr17        7675079      7675079        Frame_Shift_Del          DEL   \n",
       "1      chr17        7675079      7675079      Missense_Mutation          SNP   \n",
       "2      chr17        7675085      7675085      Missense_Mutation          SNP   \n",
       "\n",
       "  reference_allele tumor_seq_allele1 tumor_seq_allele2 dbsnp_rs  \\\n",
       "0                T                 T                 -    novel   \n",
       "1                T                 T                 G            \n",
       "2                C                 C                 G            \n",
       "\n",
       "                     sequencer group_concat(t_depth, ',')  \\\n",
       "0  Illumina Genome Analyzer II                        109   \n",
       "1  Illumina Genome Analyzer II                      99,86   \n",
       "2  Illumina Genome Analyzer II                 96,115,115   \n",
       "\n",
       "  group_concat(t_ref_count, ',') group_concat(t_alt_count, ',')  \\\n",
       "0                             76                             32   \n",
       "1                          52,58                          16,25   \n",
       "2                       50,66,66                       40,45,45   \n",
       "\n",
       "  group_concat(n_depth, ',')   group_concat(caller, '|')  \n",
       "0                        145                     varscan  \n",
       "1                    145,167       varscan|somaticsniper  \n",
       "2                131,153,153  varscan|somaticsniper|muse  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gdc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start_position</th>\n",
       "      <th>end_position</th>\n",
       "      <th>variant_classification</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>reference_allele</th>\n",
       "      <th>tumor_seq_allele1</th>\n",
       "      <th>tumor_seq_allele2</th>\n",
       "      <th>dbsnp_rs</th>\n",
       "      <th>t_depth</th>\n",
       "      <th>t_ref_count</th>\n",
       "      <th>t_alt_count</th>\n",
       "      <th>n_depth</th>\n",
       "      <th>centers</th>\n",
       "      <th>ncallers</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>17</td>\n",
       "      <td>7675079</td>\n",
       "      <td>7675085</td>\n",
       "      <td>Frame_Shift_Del</td>\n",
       "      <td>DEL</td>\n",
       "      <td>TGGGGGC</td>\n",
       "      <td>TGGGGGC</td>\n",
       "      <td>GGGGGG</td>\n",
       "      <td>novel</td>\n",
       "      <td>67</td>\n",
       "      <td>37</td>\n",
       "      <td>30</td>\n",
       "      <td>69</td>\n",
       "      <td>PINDEL|MUSE*|SOMATICSNIPER*|VARSCANI*|INDELOCA...</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome  start_position  end_position variant_classification  \\\n",
       "0         17         7675079       7675085        Frame_Shift_Del   \n",
       "\n",
       "  variant_type reference_allele tumor_seq_allele1 tumor_seq_allele2 dbsnp_rs  \\\n",
       "0          DEL          TGGGGGC           TGGGGGC            GGGGGG    novel   \n",
       "\n",
       "  t_depth t_ref_count t_alt_count n_depth  \\\n",
       "0      67          37          30      69   \n",
       "\n",
       "                                             centers ncallers  \n",
       "0  PINDEL|MUSE*|SOMATICSNIPER*|VARSCANI*|INDELOCA...        7  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_mc3"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Dump all TP53 and APC variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_dump_mc3 = pd.read_sql(\n",
    "'''\n",
    "SELECT *\n",
    "FROM mc3_selected\n",
    "WHERE hugo_symbol IN ('TP53', 'APC')\n",
    "''', conn).sort_values(['hugo_symbol', 'tumor_sample_barcode', 'start_position', 'end_position'])\n",
    "\n",
    "df_dump_gdc = pd.read_sql(\n",
    "'''\n",
    "SELECT *\n",
    "FROM gdc_grouped_callers\n",
    "WHERE hugo_symbol IN ('TP53', 'APC')\n",
    "''', conn).sort_values(['hugo_symbol', 'tumor_sample_barcode', 'start_position', 'end_position'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_dump_mc3.to_csv('../mc3_tp53_apc_only.maf')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_dump_gdc.to_csv('../gdc_tp53_apc_only.maf')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
