{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sqlite3\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<sqlite3.Cursor at 0x105143ab0>"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "conn = sqlite3.connect('../variants.sqlite')\n",
    "conn.executescript('''\\\n",
    "PRAGMA cache_size=-4192000;\n",
    "PRAGMA temp_store=MEMORY;\n",
    "PRAGMA journal_mode=OFF;\n",
    "''')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_variant_class = pd.merge(\n",
    "    pd.read_sql('''\\\n",
    "SELECT variant_classification, count(*) AS 'mc3_count'\n",
    "FROM mc3_selected\n",
    "GROUP BY variant_classification\n",
    "ORDER BY mc3_count DESC\n",
    "''', conn),\n",
    "    pd.read_sql('''\\\n",
    "SELECT variant_classification, count(*) AS 'gdc_count'\n",
    "FROM gdc_grouped_callers\n",
    "GROUP BY variant_classification\n",
    "ORDER BY gdc_count DESC\n",
    "''', conn),\n",
    "    how='outer',\n",
    "    on='variant_classification'\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "variant_classification    Missense_MutationSilent3'UTRFrame_Shift_DelNon...\n",
       "mc3_count                                                            445856\n",
       "gdc_count                                                            515859\n",
       "dtype: object"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_variant_class.sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_per_tx_sample_variant_type = pd.read_sql('''\\\n",
    "WITH gdc_by_tx AS (\n",
    "    SELECT hugo_symbol, transcript_id, variant_classification, tumor_sample_barcode, count(*) AS 'gdc_count'\n",
    "    FROM gdc_grouped_callers\n",
    "    GROUP BY hugo_symbol, transcript_id, variant_classification, tumor_sample_barcode\n",
    "), mc3_by_tx AS (\n",
    "    SELECT hugo_symbol, transcript_id, variant_classification, tumor_sample_barcode, count(*) AS 'mc3_count'\n",
    "    FROM mc3_selected\n",
    "    GROUP BY hugo_symbol, transcript_id, variant_classification, tumor_sample_barcode\n",
    ")\n",
    "SELECT * FROM gdc_by_tx\n",
    "LEFT JOIN mc3_by_tx \n",
    "USING (hugo_symbol, transcript_id, variant_classification, tumor_sample_barcode)\n",
    "UNION ALL\n",
    "SELECT * FROM mc3_by_tx\n",
    "LEFT JOIN gdc_by_tx\n",
    "USING (hugo_symbol, transcript_id, variant_classification, tumor_sample_barcode)\n",
    "WHERE gdc_by_tx.gdc_count IS NULL\n",
    "''', conn)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_per_tx_sample_variant_type = (\n",
    "    df_per_tx_sample_variant_type\n",
    "    .fillna(0)\n",
    "    .assign(\n",
    "        gdc_count=lambda df: df.gdc_count.astype(int),\n",
    "        mc3_count=lambda df: df.mc3_count.astype(int))\n",
    "    .assign(\n",
    "        diff=lambda df: df.gdc_count-df.mc3_count\n",
    "    )\n",
    "    .sort_values(['diff'], ascending=False)\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hugo_symbol</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>variant_classification</th>\n",
       "      <th>tumor_sample_barcode</th>\n",
       "      <th>gdc_count</th>\n",
       "      <th>mc3_count</th>\n",
       "      <th>diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>595842</th>\n",
       "      <td>TTN</td>\n",
       "      <td>ENST00000589042</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>TCGA-CA-6717-01A-11D-1835-10</td>\n",
       "      <td>31</td>\n",
       "      <td>0</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>595727</th>\n",
       "      <td>TTN</td>\n",
       "      <td>ENST00000589042</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>TCGA-AA-A010-01A-01D-A17O-10</td>\n",
       "      <td>30</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>437991</th>\n",
       "      <td>TTN</td>\n",
       "      <td>ENST00000591111</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>TCGA-AA-A010-01A-01D-A17O-10</td>\n",
       "      <td>29</td>\n",
       "      <td>0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>438110</th>\n",
       "      <td>TTN</td>\n",
       "      <td>ENST00000591111</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>TCGA-CA-6717-01A-11D-1835-10</td>\n",
       "      <td>29</td>\n",
       "      <td>0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>438021</th>\n",
       "      <td>TTN</td>\n",
       "      <td>ENST00000591111</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>TCGA-AN-A046-01A-21W-A050-09</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       hugo_symbol    transcript_id variant_classification  \\\n",
       "595842         TTN  ENST00000589042      Missense_Mutation   \n",
       "595727         TTN  ENST00000589042      Missense_Mutation   \n",
       "437991         TTN  ENST00000591111      Missense_Mutation   \n",
       "438110         TTN  ENST00000591111      Missense_Mutation   \n",
       "438021         TTN  ENST00000591111      Missense_Mutation   \n",
       "\n",
       "                tumor_sample_barcode  gdc_count  mc3_count  diff  \n",
       "595842  TCGA-CA-6717-01A-11D-1835-10         31          0    31  \n",
       "595727  TCGA-AA-A010-01A-01D-A17O-10         30          0    30  \n",
       "437991  TCGA-AA-A010-01A-01D-A17O-10         29          0    29  \n",
       "438110  TCGA-CA-6717-01A-11D-1835-10         29          0    29  \n",
       "438021  TCGA-AN-A046-01A-21W-A050-09         28          0    28  "
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_per_tx_sample_variant_type.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_per_tx_sample = (\n",
    "    df_per_tx_sample_variant_type\n",
    "    .groupby(['hugo_symbol', 'transcript_id', 'tumor_sample_barcode'])\n",
    "    .sum()\n",
    "    .sort_values(['diff'], ascending=False)\n",
    "    .reset_index()\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hugo_symbol</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>tumor_sample_barcode</th>\n",
       "      <th>gdc_count</th>\n",
       "      <th>mc3_count</th>\n",
       "      <th>diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TTN</td>\n",
       "      <td>ENST00000589042</td>\n",
       "      <td>TCGA-CA-6717-01A-11D-1835-10</td>\n",
       "      <td>69</td>\n",
       "      <td>0</td>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TTN</td>\n",
       "      <td>ENST00000591111</td>\n",
       "      <td>TCGA-CA-6717-01A-11D-1835-10</td>\n",
       "      <td>68</td>\n",
       "      <td>0</td>\n",
       "      <td>68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TTN</td>\n",
       "      <td>ENST00000591111</td>\n",
       "      <td>TCGA-AA-A010-01A-01D-A17O-10</td>\n",
       "      <td>48</td>\n",
       "      <td>0</td>\n",
       "      <td>48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TTN</td>\n",
       "      <td>ENST00000589042</td>\n",
       "      <td>TCGA-AA-A010-01A-01D-A17O-10</td>\n",
       "      <td>48</td>\n",
       "      <td>0</td>\n",
       "      <td>48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TTN</td>\n",
       "      <td>ENST00000591111</td>\n",
       "      <td>TCGA-AN-A046-01A-21W-A050-09</td>\n",
       "      <td>45</td>\n",
       "      <td>0</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  hugo_symbol    transcript_id          tumor_sample_barcode  gdc_count  \\\n",
       "0         TTN  ENST00000589042  TCGA-CA-6717-01A-11D-1835-10         69   \n",
       "1         TTN  ENST00000591111  TCGA-CA-6717-01A-11D-1835-10         68   \n",
       "2         TTN  ENST00000591111  TCGA-AA-A010-01A-01D-A17O-10         48   \n",
       "3         TTN  ENST00000589042  TCGA-AA-A010-01A-01D-A17O-10         48   \n",
       "4         TTN  ENST00000591111  TCGA-AN-A046-01A-21W-A050-09         45   \n",
       "\n",
       "   mc3_count  diff  \n",
       "0          0    69  \n",
       "1          0    68  \n",
       "2          0    48  \n",
       "3          0    48  \n",
       "4          0    45  "
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_per_tx_sample.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_per_tx = (\n",
    "    df_per_tx_sample.groupby(['hugo_symbol', 'transcript_id'])\n",
    "    .sum()\n",
    "    .sort_values(['diff'], ascending=False)\n",
    "    .reset_index()\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hugo_symbol</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gdc_count</th>\n",
       "      <th>mc3_count</th>\n",
       "      <th>diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TTN</td>\n",
       "      <td>ENST00000591111</td>\n",
       "      <td>1901</td>\n",
       "      <td>0</td>\n",
       "      <td>1901</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TTN</td>\n",
       "      <td>ENST00000589042</td>\n",
       "      <td>1670</td>\n",
       "      <td>0</td>\n",
       "      <td>1670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>APC</td>\n",
       "      <td>ENST00000257430</td>\n",
       "      <td>616</td>\n",
       "      <td>0</td>\n",
       "      <td>616</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>APC</td>\n",
       "      <td>ENST00000457016</td>\n",
       "      <td>555</td>\n",
       "      <td>0</td>\n",
       "      <td>555</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OBSCN</td>\n",
       "      <td>ENST00000570156</td>\n",
       "      <td>386</td>\n",
       "      <td>0</td>\n",
       "      <td>386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>DST</td>\n",
       "      <td>ENST00000312431</td>\n",
       "      <td>379</td>\n",
       "      <td>0</td>\n",
       "      <td>379</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>OBSCN</td>\n",
       "      <td>ENST00000422127</td>\n",
       "      <td>378</td>\n",
       "      <td>0</td>\n",
       "      <td>378</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>MUC4</td>\n",
       "      <td>ENST00000463781</td>\n",
       "      <td>493</td>\n",
       "      <td>131</td>\n",
       "      <td>362</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>TP53</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>1306</td>\n",
       "      <td>977</td>\n",
       "      <td>329</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>DST</td>\n",
       "      <td>ENST00000244364</td>\n",
       "      <td>324</td>\n",
       "      <td>0</td>\n",
       "      <td>324</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>NEB</td>\n",
       "      <td>ENST00000172853</td>\n",
       "      <td>309</td>\n",
       "      <td>0</td>\n",
       "      <td>309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>SSPO</td>\n",
       "      <td>ENST00000378016</td>\n",
       "      <td>300</td>\n",
       "      <td>2</td>\n",
       "      <td>298</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>NEB</td>\n",
       "      <td>ENST00000397345</td>\n",
       "      <td>293</td>\n",
       "      <td>0</td>\n",
       "      <td>293</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>FAT3</td>\n",
       "      <td>ENST00000525166</td>\n",
       "      <td>291</td>\n",
       "      <td>0</td>\n",
       "      <td>291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>MACF1</td>\n",
       "      <td>ENST00000372915</td>\n",
       "      <td>274</td>\n",
       "      <td>0</td>\n",
       "      <td>274</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>CSMD1</td>\n",
       "      <td>ENST00000520002</td>\n",
       "      <td>265</td>\n",
       "      <td>0</td>\n",
       "      <td>265</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>SYNE2</td>\n",
       "      <td>ENST00000344113</td>\n",
       "      <td>260</td>\n",
       "      <td>0</td>\n",
       "      <td>260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>CSMD1</td>\n",
       "      <td>ENST00000537824</td>\n",
       "      <td>260</td>\n",
       "      <td>0</td>\n",
       "      <td>260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>FAT3</td>\n",
       "      <td>ENST00000298047</td>\n",
       "      <td>257</td>\n",
       "      <td>0</td>\n",
       "      <td>257</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>DNAH11</td>\n",
       "      <td>ENST00000409508</td>\n",
       "      <td>256</td>\n",
       "      <td>0</td>\n",
       "      <td>256</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   hugo_symbol    transcript_id  gdc_count  mc3_count  diff\n",
       "0          TTN  ENST00000591111       1901          0  1901\n",
       "1          TTN  ENST00000589042       1670          0  1670\n",
       "2          APC  ENST00000257430        616          0   616\n",
       "3          APC  ENST00000457016        555          0   555\n",
       "4        OBSCN  ENST00000570156        386          0   386\n",
       "5          DST  ENST00000312431        379          0   379\n",
       "6        OBSCN  ENST00000422127        378          0   378\n",
       "7         MUC4  ENST00000463781        493        131   362\n",
       "8         TP53  ENST00000269305       1306        977   329\n",
       "9          DST  ENST00000244364        324          0   324\n",
       "10         NEB  ENST00000172853        309          0   309\n",
       "11        SSPO  ENST00000378016        300          2   298\n",
       "12         NEB  ENST00000397345        293          0   293\n",
       "13        FAT3  ENST00000525166        291          0   291\n",
       "14       MACF1  ENST00000372915        274          0   274\n",
       "15       CSMD1  ENST00000520002        265          0   265\n",
       "16       SYNE2  ENST00000344113        260          0   260\n",
       "17       CSMD1  ENST00000537824        260          0   260\n",
       "18        FAT3  ENST00000298047        257          0   257\n",
       "19      DNAH11  ENST00000409508        256          0   256"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_per_tx.head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Export data to excel\n",
    "df_per_tx.to_csv('../variant_count_diff.per_tx.csv', index=False)\n",
    "df_per_tx_sample.to_csv('../variant_count_diff.per_tx_sample.csv', index=False)\n",
    "df_per_tx_sample_variant_type.to_csv('../variant_count_diff.per_tx_sample_type.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hugo_symbol</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>tumor_sample_barcode</th>\n",
       "      <th>gdc_count</th>\n",
       "      <th>mc3_count</th>\n",
       "      <th>diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>APC</td>\n",
       "      <td>ENST00000457016</td>\n",
       "      <td>TCGA-CA-6717-01A-11D-1835-10</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126</th>\n",
       "      <td>APC</td>\n",
       "      <td>ENST00000257430</td>\n",
       "      <td>TCGA-CA-6717-01A-11D-1835-10</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304</th>\n",
       "      <td>APC</td>\n",
       "      <td>ENST00000257430</td>\n",
       "      <td>TCGA-AA-A010-01A-01D-A17O-10</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306</th>\n",
       "      <td>APC</td>\n",
       "      <td>ENST00000257430</td>\n",
       "      <td>TCGA-AN-A046-01A-21W-A050-09</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>309</th>\n",
       "      <td>APC</td>\n",
       "      <td>ENST00000257430</td>\n",
       "      <td>TCGA-CM-6166-01A-11D-1650-10</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>319</th>\n",
       "      <td>APC</td>\n",
       "      <td>ENST00000457016</td>\n",
       "      <td>TCGA-CM-6166-01A-11D-1650-10</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>323</th>\n",
       "      <td>APC</td>\n",
       "      <td>ENST00000457016</td>\n",
       "      <td>TCGA-AN-A046-01A-21W-A050-09</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328</th>\n",
       "      <td>APC</td>\n",
       "      <td>ENST00000457016</td>\n",
       "      <td>TCGA-AA-A010-01A-01D-A17O-10</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>503</th>\n",
       "      <td>APC</td>\n",
       "      <td>ENST00000457016</td>\n",
       "      <td>TCGA-A6-2675-01A-02D-1719-10</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>801</th>\n",
       "      <td>APC</td>\n",
       "      <td>ENST00000257430</td>\n",
       "      <td>TCGA-SS-A7HO-01A-21D-A36X-10</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    hugo_symbol    transcript_id          tumor_sample_barcode  gdc_count  \\\n",
       "89          APC  ENST00000457016  TCGA-CA-6717-01A-11D-1835-10         10   \n",
       "126         APC  ENST00000257430  TCGA-CA-6717-01A-11D-1835-10          9   \n",
       "304         APC  ENST00000257430  TCGA-AA-A010-01A-01D-A17O-10          7   \n",
       "306         APC  ENST00000257430  TCGA-AN-A046-01A-21W-A050-09          7   \n",
       "309         APC  ENST00000257430  TCGA-CM-6166-01A-11D-1650-10          7   \n",
       "319         APC  ENST00000457016  TCGA-CM-6166-01A-11D-1650-10          7   \n",
       "323         APC  ENST00000457016  TCGA-AN-A046-01A-21W-A050-09          7   \n",
       "328         APC  ENST00000457016  TCGA-AA-A010-01A-01D-A17O-10          7   \n",
       "503         APC  ENST00000457016  TCGA-A6-2675-01A-02D-1719-10          6   \n",
       "801         APC  ENST00000257430  TCGA-SS-A7HO-01A-21D-A36X-10          5   \n",
       "\n",
       "     mc3_count  diff  \n",
       "89           0    10  \n",
       "126          0     9  \n",
       "304          0     7  \n",
       "306          0     7  \n",
       "309          0     7  \n",
       "319          0     7  \n",
       "323          0     7  \n",
       "328          0     7  \n",
       "503          0     6  \n",
       "801          0     5  "
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_per_tx_sample[df_per_tx_sample['transcript_id'].isin(['ENST00000257430', 'ENST00000457016'])].head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## TP53"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Check if there is any flag for the variants of TP53."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hugo_symbol</th>\n",
       "      <th>entrez_gene_id</th>\n",
       "      <th>center</th>\n",
       "      <th>ncbi_build</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start_position</th>\n",
       "      <th>end_position</th>\n",
       "      <th>strand</th>\n",
       "      <th>variant_classification</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>...</th>\n",
       "      <th>src_vcf_id</th>\n",
       "      <th>tumor_bam_uuid</th>\n",
       "      <th>normal_bam_uuid</th>\n",
       "      <th>case_id</th>\n",
       "      <th>gdc_filter</th>\n",
       "      <th>cosmic</th>\n",
       "      <th>mc3_overlap</th>\n",
       "      <th>gdc_validation_status</th>\n",
       "      <th>cancer_type</th>\n",
       "      <th>group_concat(caller, '|')</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TP53</td>\n",
       "      <td>7157</td>\n",
       "      <td>WUGSC</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>chr17</td>\n",
       "      <td>7675124</td>\n",
       "      <td>7675124</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>d410118f-5ac1-4487-9e5b-6a91915e4a7b</td>\n",
       "      <td>fe8f1f10-e7c5-4faa-815e-c0f7b9d78767</td>\n",
       "      <td>fc0e1648-2ae8-4f56-8ebb-c93e3b1c48f5</td>\n",
       "      <td>b8023162-5e82-40e6-ad8c-8acf81821f01</td>\n",
       "      <td></td>\n",
       "      <td>COSM10808;COSM129852;COSM129853;COSM129854;COS...</td>\n",
       "      <td>True</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>OV</td>\n",
       "      <td>somaticsniper|muse|varscan|mutect</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TP53</td>\n",
       "      <td>7157</td>\n",
       "      <td>WUGSC</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>chr17</td>\n",
       "      <td>7675229</td>\n",
       "      <td>7675229</td>\n",
       "      <td>+</td>\n",
       "      <td>Frame_Shift_Del</td>\n",
       "      <td>DEL</td>\n",
       "      <td>...</td>\n",
       "      <td>2318c30f-26cc-48a0-a5e3-86398c54d979</td>\n",
       "      <td>fe8f1f10-e7c5-4faa-815e-c0f7b9d78767</td>\n",
       "      <td>fc0e1648-2ae8-4f56-8ebb-c93e3b1c48f5</td>\n",
       "      <td>b8023162-5e82-40e6-ad8c-8acf81821f01</td>\n",
       "      <td></td>\n",
       "      <td>COSM5198771;COSM5198772;COSM5198773;COSM519877...</td>\n",
       "      <td>True</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>OV</td>\n",
       "      <td>varscan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TP53</td>\n",
       "      <td>7157</td>\n",
       "      <td>WUGSC</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>chr17</td>\n",
       "      <td>7675232</td>\n",
       "      <td>7675232</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>3c685c63-6618-403d-979f-f5a69b9d805b</td>\n",
       "      <td>fe8f1f10-e7c5-4faa-815e-c0f7b9d78767</td>\n",
       "      <td>fc0e1648-2ae8-4f56-8ebb-c93e3b1c48f5</td>\n",
       "      <td>b8023162-5e82-40e6-ad8c-8acf81821f01</td>\n",
       "      <td></td>\n",
       "      <td>COSM3403294;COSM3403295;COSM3403296;COSM340329...</td>\n",
       "      <td>True</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>OV</td>\n",
       "      <td>somaticsniper</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TP53</td>\n",
       "      <td>7157</td>\n",
       "      <td>WUGSC</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>chr17</td>\n",
       "      <td>7675232</td>\n",
       "      <td>7675232</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>2318c30f-26cc-48a0-a5e3-86398c54d979</td>\n",
       "      <td>fe8f1f10-e7c5-4faa-815e-c0f7b9d78767</td>\n",
       "      <td>fc0e1648-2ae8-4f56-8ebb-c93e3b1c48f5</td>\n",
       "      <td>b8023162-5e82-40e6-ad8c-8acf81821f01</td>\n",
       "      <td></td>\n",
       "      <td>COSM3403294;COSM3403295;COSM3403296;COSM340329...</td>\n",
       "      <td>True</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>OV</td>\n",
       "      <td>varscan</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4 rows × 122 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  hugo_symbol entrez_gene_id center ncbi_build chromosome start_position  \\\n",
       "0        TP53           7157  WUGSC     GRCh38      chr17        7675124   \n",
       "1        TP53           7157  WUGSC     GRCh38      chr17        7675229   \n",
       "2        TP53           7157  WUGSC     GRCh38      chr17        7675232   \n",
       "3        TP53           7157  WUGSC     GRCh38      chr17        7675232   \n",
       "\n",
       "  end_position strand variant_classification variant_type  \\\n",
       "0      7675124      +      Missense_Mutation          SNP   \n",
       "1      7675229      +        Frame_Shift_Del          DEL   \n",
       "2      7675232      +      Missense_Mutation          SNP   \n",
       "3      7675232      +      Missense_Mutation          SNP   \n",
       "\n",
       "                 ...                                            src_vcf_id  \\\n",
       "0                ...                  d410118f-5ac1-4487-9e5b-6a91915e4a7b   \n",
       "1                ...                  2318c30f-26cc-48a0-a5e3-86398c54d979   \n",
       "2                ...                  3c685c63-6618-403d-979f-f5a69b9d805b   \n",
       "3                ...                  2318c30f-26cc-48a0-a5e3-86398c54d979   \n",
       "\n",
       "                         tumor_bam_uuid                       normal_bam_uuid  \\\n",
       "0  fe8f1f10-e7c5-4faa-815e-c0f7b9d78767  fc0e1648-2ae8-4f56-8ebb-c93e3b1c48f5   \n",
       "1  fe8f1f10-e7c5-4faa-815e-c0f7b9d78767  fc0e1648-2ae8-4f56-8ebb-c93e3b1c48f5   \n",
       "2  fe8f1f10-e7c5-4faa-815e-c0f7b9d78767  fc0e1648-2ae8-4f56-8ebb-c93e3b1c48f5   \n",
       "3  fe8f1f10-e7c5-4faa-815e-c0f7b9d78767  fc0e1648-2ae8-4f56-8ebb-c93e3b1c48f5   \n",
       "\n",
       "                                case_id gdc_filter  \\\n",
       "0  b8023162-5e82-40e6-ad8c-8acf81821f01              \n",
       "1  b8023162-5e82-40e6-ad8c-8acf81821f01              \n",
       "2  b8023162-5e82-40e6-ad8c-8acf81821f01              \n",
       "3  b8023162-5e82-40e6-ad8c-8acf81821f01              \n",
       "\n",
       "                                              cosmic mc3_overlap  \\\n",
       "0  COSM10808;COSM129852;COSM129853;COSM129854;COS...        True   \n",
       "1  COSM5198771;COSM5198772;COSM5198773;COSM519877...        True   \n",
       "2  COSM3403294;COSM3403295;COSM3403296;COSM340329...        True   \n",
       "3  COSM3403294;COSM3403295;COSM3403296;COSM340329...        True   \n",
       "\n",
       "  gdc_validation_status cancer_type          group_concat(caller, '|')  \n",
       "0               Unknown          OV  somaticsniper|muse|varscan|mutect  \n",
       "1               Unknown          OV                            varscan  \n",
       "2               Unknown          OV                      somaticsniper  \n",
       "3               Unknown          OV                            varscan  \n",
       "\n",
       "[4 rows x 122 columns]"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample = 'TCGA-WR-A838-01A-12D-A403-09'\n",
    "gene = 'TP53'\n",
    "pd.read_sql(\n",
    "    '''\n",
    "    SELECT *\n",
    "    FROM gdc_grouped_callers\n",
    "    WHERE hugo_symbol=? AND tumor_sample_barcode=?\n",
    "    ''',\n",
    "    conn,\n",
    "    params=[gene, sample]\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
